% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
  11pt,
  a4paper,
]{book}

\usepackage{amsmath,amssymb}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\ifPDFTeX\else  
    % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\setlength{\emergencystretch}{3em} % prevent overfull lines
\setcounter{secnumdepth}{5}
% Make \paragraph and \subparagraph free-standing
\makeatletter
\ifx\paragraph\undefined\else
  \let\oldparagraph\paragraph
  \renewcommand{\paragraph}{
    \@ifstar
      \xxxParagraphStar
      \xxxParagraphNoStar
  }
  \newcommand{\xxxParagraphStar}[1]{\oldparagraph*{#1}\mbox{}}
  \newcommand{\xxxParagraphNoStar}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
  \let\oldsubparagraph\subparagraph
  \renewcommand{\subparagraph}{
    \@ifstar
      \xxxSubParagraphStar
      \xxxSubParagraphNoStar
  }
  \newcommand{\xxxSubParagraphStar}[1]{\oldsubparagraph*{#1}\mbox{}}
  \newcommand{\xxxSubParagraphNoStar}[1]{\oldsubparagraph{#1}\mbox{}}
\fi
\makeatother

\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{241,243,245}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.68,0.00,0.00}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{#1}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.40,0.45,0.13}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.68,0.00,0.00}{#1}}
\newcommand{\BuiltInTok}[1]{\textcolor[rgb]{0.00,0.23,0.31}{#1}}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.13,0.47,0.30}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{#1}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{\textit{#1}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.00,0.23,0.31}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.68,0.00,0.00}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.68,0.00,0.00}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{\textit{#1}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.68,0.00,0.00}{#1}}
\newcommand{\ExtensionTok}[1]{\textcolor[rgb]{0.00,0.23,0.31}{#1}}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.68,0.00,0.00}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.28,0.35,0.67}{#1}}
\newcommand{\ImportTok}[1]{\textcolor[rgb]{0.00,0.46,0.62}{#1}}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{#1}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.00,0.23,0.31}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{\textcolor[rgb]{0.00,0.23,0.31}{#1}}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{#1}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.00,0.23,0.31}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.68,0.00,0.00}{#1}}
\newcommand{\RegionMarkerTok}[1]{\textcolor[rgb]{0.00,0.23,0.31}{#1}}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.13,0.47,0.30}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.13,0.47,0.30}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.07,0.07,0.07}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.13,0.47,0.30}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{\textit{#1}}}

\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
% definitions for citeproc citations
\NewDocumentCommand\citeproctext{}{}
\NewDocumentCommand\citeproc{mm}{%
  \begingroup\def\citeproctext{#2}\cite{#1}\endgroup}
\makeatletter
 % allow citations to break across lines
 \let\@cite@ofmt\@firstofone
 % avoid brackets around text for \cite:
 \def\@biblabel#1{}
 \def\@cite#1#2{{#1\if@tempswa , #2\fi}}
\makeatother
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newlength{\csllabelwidth}
\setlength{\csllabelwidth}{3em}
\newenvironment{CSLReferences}[2] % #1 hanging-indent, #2 entry-spacing
 {\begin{list}{}{%
  \setlength{\itemindent}{0pt}
  \setlength{\leftmargin}{0pt}
  \setlength{\parsep}{0pt}
  % turn on hanging indent if param 1 is 1
  \ifodd #1
   \setlength{\leftmargin}{\cslhangindent}
   \setlength{\itemindent}{-1\cslhangindent}
  \fi
  % set entry spacing
  \setlength{\itemsep}{#2\baselineskip}}}
 {\end{list}}
\usepackage{calc}
\newcommand{\CSLBlock}[1]{\hfill\break\parbox[t]{\linewidth}{\strut\ignorespaces#1\strut}}
\newcommand{\CSLLeftMargin}[1]{\parbox[t]{\csllabelwidth}{\strut#1\strut}}
\newcommand{\CSLRightInline}[1]{\parbox[t]{\linewidth - \csllabelwidth}{\strut#1\strut}}
\newcommand{\CSLIndent}[1]{\hspace{\cslhangindent}#1}

\makeatletter
\@ifpackageloaded{tcolorbox}{}{\usepackage[skins,breakable]{tcolorbox}}
\@ifpackageloaded{fontawesome5}{}{\usepackage{fontawesome5}}
\definecolor{quarto-callout-color}{HTML}{909090}
\definecolor{quarto-callout-note-color}{HTML}{0758E5}
\definecolor{quarto-callout-important-color}{HTML}{CC1914}
\definecolor{quarto-callout-warning-color}{HTML}{EB9113}
\definecolor{quarto-callout-tip-color}{HTML}{00A047}
\definecolor{quarto-callout-caution-color}{HTML}{FC5300}
\definecolor{quarto-callout-color-frame}{HTML}{acacac}
\definecolor{quarto-callout-note-color-frame}{HTML}{4582ec}
\definecolor{quarto-callout-important-color-frame}{HTML}{d9534f}
\definecolor{quarto-callout-warning-color-frame}{HTML}{f0ad4e}
\definecolor{quarto-callout-tip-color-frame}{HTML}{02b875}
\definecolor{quarto-callout-caution-color-frame}{HTML}{fd7e14}
\makeatother
\makeatletter
\@ifpackageloaded{caption}{}{\usepackage{caption}}
\AtBeginDocument{%
\ifdefined\contentsname
  \renewcommand*\contentsname{Table of contents}
\else
  \newcommand\contentsname{Table of contents}
\fi
\ifdefined\listfigurename
  \renewcommand*\listfigurename{List of Figures}
\else
  \newcommand\listfigurename{List of Figures}
\fi
\ifdefined\listtablename
  \renewcommand*\listtablename{List of Tables}
\else
  \newcommand\listtablename{List of Tables}
\fi
\ifdefined\figurename
  \renewcommand*\figurename{Figure}
\else
  \newcommand\figurename{Figure}
\fi
\ifdefined\tablename
  \renewcommand*\tablename{Table}
\else
  \newcommand\tablename{Table}
\fi
}
\@ifpackageloaded{float}{}{\usepackage{float}}
\floatstyle{ruled}
\@ifundefined{c@chapter}{\newfloat{codelisting}{h}{lop}}{\newfloat{codelisting}{h}{lop}[chapter]}
\floatname{codelisting}{Listing}
\newcommand*\listoflistings{\listof{codelisting}{List of Listings}}
\makeatother
\makeatletter
\makeatother
\makeatletter
\@ifpackageloaded{caption}{}{\usepackage{caption}}
\@ifpackageloaded{subcaption}{}{\usepackage{subcaption}}
\makeatother

\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\usepackage{bookmark}

\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same} % disable monospaced font for URLs
\hypersetup{
  pdftitle={DD2 biobank},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}


\title{DD2 biobank}
\usepackage{etoolbox}
\makeatletter
\providecommand{\subtitle}[1]{% add subtitle to \maketitle
  \apptocmd{\@title}{\par {\large #1 \par}}{}{}
}
\makeatother
\subtitle{Primary data derived from DD2 research studies}
\author{}
\date{}

\begin{document}
\frontmatter
\maketitle

\renewcommand*\contentsname{Table of contents}
{
\setcounter{tocdepth}{2}
\tableofcontents
}

\mainmatter
Go to \href{dd2_biobank.qmd\#data-documentation}{Data documentation}

When individuals are enrolled in the DD2, blood and urine samples are
collected and stored in the biobank in Vejle. The samples themselves are
considered ``primary DD2 enrollment data'', and they are all collected
at DD2 \emph{baseline}. No automatic or standard analyses are conducted,
but DD2 research projects can have the samples analyzed if (additional)
analyses are needed - the timing for the analysis results will always be
baseline DD2, because it is the time the blood/urine sample was
collected. The results from analyses of the blood and urine samples are
considered ``Additional DD2 data''.

For further information about the initial idea about the biobank, please
see Christensen et al.\\
The documentation for the biobank can be downloaded here (Danish):

\chapter{Data sources}\label{data-sources}

The data in the biobank are joined from multiple studies and datasets.
The individual identifier in the biobank is \texttt{ProjektID} and it is
unique per CPR number. The project ID should end with the digits -00
(individual). Another variable is \texttt{Barcode} which is similar to
the project id, but it end with a number (e.g.~-12, -19, or -99) and
denotes the specific sample for the individual (see documentation
above).

Locally at DCE, the (original) data files are stored in the folder:\\
O:\textbackslash HE\_KEA-DATA-RAW0050\textbackslash Main
Part\textbackslash data\textbackslash Input Data
Sets\textbackslash BioBank

Data files from the 2022/2023 rounds are stored in the folder:\\
O:\textbackslash HE\textbackslash KEA-DATA-RAW0050\textbackslash Biomarkører

Below is a list of the biobank data files stored at DCE (files read in
locally). File names might have changed since we initially received
them.

\section{First biomarkers}\label{first-biomarkers}

In the first phase of DD2, a lot of biomarkers were analysed for the
first 1,053 individuals enrolled in DD2 (some individuals have later
opted out). These individuals are marked by the variable
\texttt{BlodProve1053patients} (based on the CPR numbers in data file
\texttt{First1053Patients} (same as in
\texttt{UrinResultatermedRatio2013Nov})) and are predominantly enrolled
in 2011-2012.

\begin{itemize}
\item
  \texttt{First1053Patients.txt}: The file appears to have been received
  in November 2012. It includes n=1,053 CPR numbers and these
  individuals are most likely everyone in the database at that time. The
  file includes results from the initial blood samples. The dataset
  lacks date information and also unit specifications for the variables
  hæmolyse, icteri, and lipæmi. The dataset includes the variables:

  \begin{itemize}
  \tightlist
  \item
    CPR
  \item
    ProjektID
  \item
    C-peptid (n=1,053, pmol/L)
  \item
    GAD (n=1,053, kU/l)
  \item
    Glucose (n=1,050, mmol/L)
  \item
    ALAT (n=1,030, U/L)
  \item
    Hæmolyse (n=1,030)
  \item
    Icteri (n=1,030)
  \item
    Lipæmi (n=1,030)
  \item
    AMYLP (n=1,041, U/L)
  \item
    CRP (n=1,041, mg/L)
  \end{itemize}
\end{itemize}

We have been told (e-mail from JSN 31.05.2018) not to use the c-peptide
measurements from this file (old analysis kit).

Information about some of the variables can be found in Mor et al.
(2014).

\begin{itemize}
\item
  \texttt{UrinResultatermedRatio2013Nov.txt}: We likely received the
  data file in December 2013. It includes n=1,053 CPR numbers (same as
  in \texttt{First1053Patients}) with results from the initial urine
  samples. The dataset includes the variables:

  \begin{itemize}
  \tightlist
  \item
    CPR
  \item
    Barkode (ends with -19)
  \item
    ALBu\_mg\_L\_ (n=1,041, mg/L, dated November 19, 20, and 23, 2012)
  \item
    KREA\_mmol\_L\_ (n=1,041, mmol/L, dated November 19, 20, and 23,
    2012)
  \item
    UPROT\_g\_L\_ (n=1,041, g/L, dated November 19, 20, and 23, 2012)
  \item
    Albumin\_Kreatinin\_ratio (n=1,041)
  \item
    Kommentar (n=473, \textless5 with ``\emph{PROT-U \textgreater{} 2.5
    g/l}'' and the rest with ``\emph{ALB-U \textless3 mg/L}'')
  \end{itemize}
\end{itemize}

\section{Additional biomarkers}\label{additional-biomarkers}

\begin{itemize}
\item
  \texttt{DataTilDD2medFastende.txt}: We probably received the file in
  September 2015 (Anne Gedebjerg). It includes n=5,996 CPR numbers with
  character results on GAD, glucose, and c-peptide, along with a
  variable about faste. The majority of the individuals in the file are
  enrolled before 2015, but it is only around 80-85\% of all the
  individuals enrolled before 2016. There are no dates in the data. The
  dataset includes the variables:

  \begin{itemize}
  \tightlist
  \item
    Id (ProjectID, ends with -00)
  \item
    GAD (numeric values for n=131 ID numbers, the value
    ``\emph{\textless0,000}'' for n=5,797 ID numbers,
    ``\emph{\textgreater525,000}'' for n=51 ID numbers, and
    ``\emph{?????}'' or ``\emph{NA}'' for the remaining n=20 ID numbers)
  \item
    Glucose (numeric values for n=4,457 and ``\emph{NA}'' for the
    remaining n=1,539 ID numbers)
  \item
    Cpeptid (numeric values for n=5,964 and ``\emph{NA}'' for the
    remaining n=32 ID numbers)
  \item
    Fastende (``\emph{Ja}'' for n=2,891, ``\emph{Nej}'' for n=397, and
    ``\emph{Ved ikke}'' for the remaining n=2,708 ID numbers) (see below
    for more information about fasting blood samples)
  \end{itemize}
\end{itemize}

The majority of the initial 1,053 individuals are also included in this
data file. We have been told to use the c-peptide measurements in this
data file and not the original data file.

\begin{itemize}
\tightlist
\item
  \texttt{WrongCPeptideMeasurements.txt}: The file appears to have been
  received in November 2015. It includes n=105 ID numbers (end with -00)
  and the variables cpeptid and NyCpeptid. The values are quite
  different. We have noted in the syntax that JSN has told us not to use
  these variables (e-mail 31.05.2018).
\end{itemize}

\section{Results from Anne Gedebjerg}\label{results-from-anne-gedebjerg}

\begin{itemize}
\item
  \texttt{DD2\ resultater.xlsx} (and \texttt{DD2\ resultater\_10\_0095},
  which is the version where \textless10 is replaced by 10 and
  \textless0,095 by 0,095): We have received the file in September 2017
  (from Anne Gedebjerg). It includes n=7,519 barcodes (end with -99) in
  sheets of 100 or 101 rows, and should include CRP and mannose-binding
  lectin (MBL) on everyone enrolled by December 2016. See Gedebjerg et
  al. (2023) for more information. The dataset includes the variables:

  \begin{itemize}
  \tightlist
  \item
    barkode (ends with -99)
  \item
    CRP (n=7,510, mg/L)
  \item
    MBL (n=7,514, µg/L)
  \end{itemize}
\end{itemize}

We have been told to keep the CRP measurements from the original file
and this file separate. The unit for CRP is mg/L in both data files.

\begin{itemize}
\item
  \texttt{Resultater\ den\ 250917\ Anne\ Gedebjerg.xlsx}: We have
  received the file in September 2017 (from Anne Gedebjerg). It includes
  n=3,116 barcodes (end with -99) and variables regarding MBL expression
  genotyping (six SNPs in the MBL2 gene). The genotyping was done for
  the first \textasciitilde3,000 individuals enrolled in DD2. See
  Gedebjerg et al. (2020) for more information. The dataset includes the
  variables:

  \begin{itemize}
  \tightlist
  \item
    barkode (ends with -99)
  \item
    HL
  \item
    XY
  \item
    PQ
  \item
    52
  \item
    54
  \item
    57
  \item
    HAPLOTYPE
  \end{itemize}
\end{itemize}

\section{April 2022 data}\label{april-2022-data}

During 2022-2023 we received additional data on CRP, c-peptide, and
glucose. We had a file with CRP in October 2022, but it is fully
included in a file from January 2023 which also includes c-peptide and
glucose. The October file is therefore not used during uploads, whereas
the January 2023 file has been uploaded to the servers.

\begin{itemize}
\item
  \texttt{DD2\_cRP\_Glucose\_Cpep\_2022\_resultater\ (1).xlsx}: The data
  file includes n=3,399 projekt\_id. They are all enrolled after the
  first n=1,053 individuals, but we don't know why they were analysed
  (not all individuals per year. Maybe in IDA?). The dataset includes
  the variables: projekt\_id, Cpeptid\_Barkode, Cpeptid\_Resultat,
  Cpeptid\_Måleenhed, Cpeptid\_Antal\_decimaler, Cpeptid\_Dato,
  Cpeptid\_Notat, CRPHS\_Barkode, CRPHS\_Resultat, CRPHS\_Måleenhed,
  CRPHS\_Antal\_decimaler, CRPHS\_Dato, CRPHS\_Notat, Glukose\_Barkode,
  Glukose\_Resultat, Glukose\_Måleenhed, Glukose\_Antal\_decimaler,
  Glukose\_Dato, Glukose\_Notat.

  \begin{itemize}
  \tightlist
  \item
    projekt\_id (n=3,339)
  \item
    c-peptid (n=2,933, pmol/l, dated 30APR2022 or 01MAY2022)
  \item
    CRP (n=2,478, mg/l, dated 30APR2022 or 01MAY2022)
  \item
    Glucose (n=3,055, mmol/L, dated 02APR2022 or 03APR2022)
  \end{itemize}
\end{itemize}

\section{C-peptide and glucose (full)}\label{c-peptide-and-glucose-full}

During the summer 2023 we have received data on c-peptide (July 2023)
and glucose (August 2023). These files include \textbf{all} measurements
from the biobank (cleaned), and results from these files will thus
replace all the other measurements from earlier datasets. We now have
the files:

\begin{itemize}
\tightlist
\item
  \texttt{dd2\_all\_C\_peptide\_14July2023.xls}: Includes n=9,762
  project\_id with data on c-peptide. The file also includes information
  about analysis date, sampling date, freezing date, unit, and kit. The
  following data management has been done by DD2 before we received the
  file:
\end{itemize}

\begin{tcolorbox}[enhanced jigsaw, toptitle=1mm, opacitybacktitle=0.6, arc=.35mm, titlerule=0mm, colbacktitle=quarto-callout-note-color!10!white, coltitle=black, bottomrule=.15mm, opacityback=0, rightrule=.15mm, bottomtitle=1mm, breakable, left=2mm, title=\textcolor{quarto-callout-note-color}{\faInfo}\hspace{0.5em}{Quote from e-mail}, colframe=quarto-callout-note-color-frame, toprule=.15mm, leftrule=.75mm, colback=white]

\begin{verbatim}
- Alle 1254 observationer, som havde"gamle" målinger (før 28. feb 2015) er erstattet med opdaterede målinger med nyt kit/assay (efter 28. feb 2015).
- 6 observationer, som kun havde en "gammel" måling og INGEN opdateret genmåling er slettet
- Der er renset op i data, så hvert individ kun fremgår med én måling, som er analyseret vha. nyt kit/assay 
\end{verbatim}

\end{tcolorbox}

Please note that c-peptide is now in another unit (it is just the old
unit divided by 1000).

\begin{itemize}
\tightlist
\item
  \texttt{DD2\_glukoser\_2023\_08\_23.xlsx}: The data file includes
  n=9,563 projekt\_id and information about glucose, units, and dates.
\end{itemize}

\subsection{HOMA}\label{homa}

\href{https://www.rdm.ox.ac.uk/about/our-clinical-facilities-and-mrc-units/DTU/software/homa/history}{HOMA}
values are calculated based on c-peptide and glucose. We use the Oxford
calculator which can be downloaded from the website:
\url{https://www.dtu.ox.ac.uk/homacalculator/download.php}. Since summer
2024, access to the HOMA calculator requires a licence (should be free
of charge for ``academic researchers'').

HOMA values in the data are calculated based on the c-peptide and
glucose measurements received during the summer 2023. HOMA has only been
estimated for glucose values in the interval 3.0-25 and c-peptide
0.2-3.5 (because of an updated Oxford HOMA calculator where values out
of range cause problems in the excel calculator). Some individuals might
therefore have glucose and c-peptide values but no HOMA values in the
new data.

\section{``Pladebiomarkører''}\label{pladebiomarkuxf8rer}

During 2022-2023, we have received new data from additional biomarker
analyses (from the Allan Vaag grant). The majority of the new biomarkers
are so-called ``pladebiomarkører'' (as opposed to the current ones which
are called ``målebiomarkører'') because of the way the analyses are
performed. In practice, everyone enrolled as of the day the blood
samples were taken from the biobank (in the beginning of 2022) were
included in the analysis (approx. the first 9,200 individuals). A small
number of individuals have multiple measurements for specific
biomarkers, most likely due to sample dilution during the analytical
process. Data are in long format and include the following 22 biomarkers
(all with unit pg/ml) and the dates refer to when we received the data
files:

\begin{itemize}
\tightlist
\item
  TNF-a (April 2022, n=9,202, pg/ml)
\item
  IL-6 (April 2022, n=9,195, pg/ml)
\item
  Ang-Like4 (November 2022, n=9,200, pg/ml)
\item
  FGF-21 (November 2022, n=9,200, pg/ml)
\item
  FGF-23 (November 2022, Hu FGF-23, n=9,200, pg/ml)
\item
  IL1-RA (November 2022, n=9,200, pg/ml)
\item
  Leptin (November 2022, n=9,196, pg/ml)
\item
  RAGE (November 2022, soluble, n=9,200, pg/ml)
\item
  Sclerostin (November 2022, n=9,200, pg/ml)
\item
  U-PAR (November 2022, n=9,200, pg/ml)
\item
  Osteocalcin-1 (February 2023, n=9,203)
\item
  CD163 (April 2023, n=9,047, pg/ml)
\item
  Galectin-3 (April 2023, n=9,008, pg/ml)
\item
  GDF-15 (April 2023, n=9,046, pg/ml)
\item
  NT-proBNP (April 2023, n=9,048, pg/ml)
\item
  Resistin (April 2023, n=9,046, pg/ml)
\item
  Serpin (April 2023, n=9,047, pg/ml)
\item
  YKL-40 (April 2023, n=9,045, pg/ml)
\item
  Osteopontin (June 2023, n=9,204, pg/ml)
\item
  Adiponectin (July 2023, n=9,204, pg/ml)
\item
  Follistatin (July 2023, n=9,204, pg/ml)
\item
  MPO (July 2023, n=9,204, pg/ml)
\end{itemize}

An overview of the biomarkers (table from the Allan Vaag grant
application) can be found here:

An additional document combining overview sheets, method descriptions,
and quality logs from some of the analysis rounds can be downloaded
here:

The data files with ``pladebiomarkører'' and ``målebiomarkører'' don't
have the same format (e.g., long vs.~wide format) and we have thus not
combined the data files.

\subsection{Data files
(pladebiomarkører)}\label{data-files-pladebiomarkuxf8rer}

In the following section we will go through the data files we have
received.

\begin{itemize}
\item
  April 2022, \texttt{220211\ Vplex\_final.xlsx} with 2 sheets (data and
  background information). Data received in April 2022 but probably
  analyzed in February 2022 (no dates in data, but based on date stamps
  in file names). The data file includes data from n=9,294 DD2
  individuals on IL-6 and TNF-a. The first sheet,
  \texttt{Vplex\ sample\ results\_final}, includes the variables: Sample
  (id, ends with -12), Sample\_Group (=Sample in all rows), Assay
  (either TNF-a or IL-6), Calc\_\_Conc\_\_Mean (results), RANGE (value 1
  or 2), Plate\_Name (each Plate\_Name is used 78 times).

  \begin{itemize}
  \tightlist
  \item
    TNF-a (n=9,202, pg/ml)
  \item
    IL-6 (n=9,195, pg/ml)
  \end{itemize}

  The second sheet, \texttt{Vplex\ complete\ final}, includes background
  information (rådata) about the sample from the sample\_groups
  \emph{Sample} (n=9,024), \emph{Standards} (n=1,888), and
  \emph{Internal Control} (n=236). The sheet includes the variables
  Plate\_Name, Sample\_Group, Sample, Assay, Well, Signal, Mean, CV
  Calc\_\_Concentration, Calc\_\_Conc\_\_Mean, Calc\_\_Conc\_\_CV,
  \_\_Recovery, \_\_Recovery\_Mean, Detection\_Limits\_\_Calc\_\_Low,
  Detection\_Limits\_\_Calc\_\_High, Detection\_Range,
  Detection\_Range\_yesno, Quantification\_range,
  Quantification\_range\_yesno, RANGE.

  We also received the data file
  \texttt{DD2\ quality\ log\_panel\ 1\ edited\_TNF\ IL6.xlsx} but is has
  not been used.
\item
  November 2022, \texttt{8plex\ data\ final.xlsx}, with 9 sheets
  (overview + 8 biomarkers). We received the data file in November 2022.
  There are no date stamps indicating when the analyses were performed.
  The data file include information on n=9,204 individuals (based on
  sample ID ending with -12). For each assay, the data file includes the
  variables sample (ID), assay, calc\_\_conc\_\_mean (result), RANGE,
  and plate\_name. In an e-mail, we have been told the unit is pg/ml for
  all assays.

  \begin{itemize}
  \tightlist
  \item
    Ang-Like4 (n=9,200, pg/ml)
  \item
    FGF-21 (n=9,200, pg/ml)
  \item
    FGF-23 (Hu FGF-23, n=9,200, pg/ml)
  \item
    IL1-RA (n=9,200, pg/ml)
  \item
    Leptin (n=9,196, pg/ml)
  \item
    RAGE (soluble, n=9,200, pg/ml)
  \item
    Sclerostin (n=9,200, pg/ml)
  \item
    U-PAR (n=9,200, pg/ml)
  \end{itemize}

  We also received data files \texttt{eight\ biomarkers\ with\ CPR.xlsx}
  and \texttt{seven\ biomarkers\ with\ CPR.xls} but these have not been
  used.
\item
  February 2023, \texttt{DD2\ osteocalcin\ final.xlsx} with 3 sheets
  (overview, results, and \emph{rådata}). We received the file in
  February 2023, but we have no indication of when the analyses were
  performed. The file includes n=9,204 individuals (sample, ends with
  -12). The data file includes the variables sample (ID), assay,
  calc\_\_conc\_\_mean (result), RANGE, and plate\_name.

  \begin{itemize}
  \tightlist
  \item
    Osteocalcin-1 (n=9,203)
  \end{itemize}

  The sheet \emph{rådata} includes detailed information about each of
  the plates.
\item
  April 2023, \texttt{DD2\_7plex\_data\_final.xlsx} with 9 sheets
  (overview + 7 biomarkers + additional sheet with sample names). We
  received the data file in April 2023. There are no date stamps
  indicating when the analyses were performed. The data file include
  information on n=9,048 individuals (based on sample ID ending with
  -12). For each assay, the data file includes the variables sample
  (ID), assay, calc\_conc\_mean (result), RANGE, and plate\_name. In an
  e-mail, we have been told the unit is pg/ml for all assays.

  \begin{itemize}
  \tightlist
  \item
    CD163 (n=9,047, pg/ml)
  \item
    Galectin-3 (n=9,008, pg/ml)
  \item
    GDF-15 (n=9,046, pg/ml)
  \item
    NT-proBNP (n=9,048, pg/ml)
  \item
    Resistin (n=9,046, pg/ml)
  \item
    Serpin (n=9,047, pg/ml)
  \item
    YKL-40 (n=9,045, pg/ml)
  \end{itemize}
\item
  June 2023, \texttt{Osteopontin\ data\ final\ (1).xlsx} with 3 sheets
  (overview + data + \emph{rådata}). We received the file in June 2023,
  but we have no indication of when the analyses were performed. It
  includes n=9,207 ID numbers (end with -12, plus a note that it means
  ``EDTA plasma fraction'') with data on osteopontin.

  \begin{itemize}
  \tightlist
  \item
    Osteopontin (n=9,204, pg/ml)
  \end{itemize}
\item
  July 2023 (1), \texttt{DD2\_adiponectin\_blue\ panel\_final\ (1).xlsx}
  with 3 sheets (overview + data + \emph{rådata}). We received the file
  in July 2023, but we have no indication of when the analyses were
  performed. It includes n=9,204 ID numbers (sample, end with -12) with
  data on adiponectin.

  \begin{itemize}
  \tightlist
  \item
    Adiponectin (n=9,204, pg/ml)
  \end{itemize}

  The sheet \emph{rådata} includes detailed information about each of
  the plates.
\item
  July 2023 (2), \texttt{DD2\ red\ panel\_final.xlsx} with 4 sheets
  (overview + data (Follistatin + MPO) + \emph{rådata}). We received the
  file in July 2023, but we have no indication of when the analyses were
  performed. It includes n=9,204 ID numbers (sample, end with -12) with
  data on follistatin and MPO

  \begin{itemize}
  \tightlist
  \item
    Follistatin (n=9,204, pg/ml)
  \item
    MPO (n=9,204, pg/ml)
  \end{itemize}

  The sheet \emph{rådata} includes detailed information about each of
  the plates.
\end{itemize}

\chapter{Fasting}\label{fasting}

Is the individual fasting? A simple question, yet, difficult to assess.

Data from the DD2 questionnaire itself include the variable
\texttt{Er\_patienten\_fastende\_} (whether the patient is fasting).
Currently, around 75\% of the individuals have answered that they are
fasting. This variable can be used on its own, but should probably be
combined with the variable \texttt{Tages\_der\_blodproeve\_i\_forbindel}
(whether the blood sample was taken at the same time as the
questionnaire was answered). If the fasting patients are restricted to
include only those whose blood sample was taken at the same time as the
questionnaire was answered, then around 72\% of the individuals are
defined to be fasting.

The data file \texttt{DataTilDD2medFastende.txt} received in September
2015 included information on fasting state for n=5,996 individuals
(``\emph{Ja}'' for n=2,891, ``\emph{Nej}'' for n=397, and ``\emph{Ved
ikke}'' for n=2,708 individuals). This file will not be updated and we
therefore don't get new information on this fasting variable. We don't
know how this variable was defined, but it is probably based on
information on the blood sample itself. By adding the information from
this file (variable name \texttt{NewFastende}) to the variable
\texttt{Er\_patienten\_fastende\_}, an additional 447 individuals can be
defined as fasting (410 with missing information in
\texttt{Er\_patienten\_fastende\_} and 37 who replied not to be fasting
in \texttt{Er\_patienten\_fastende\_}).

Currently, the fasting state has been defined by the following (SAS)
algorithm combining all the files and stating that the individual is
fasting if we have any indication that this could be the case (macro:
\texttt{AdditionalVars\_Faste}):

\begin{Shaded}
\begin{Highlighting}[]
    \ControlFlowTok{if}\NormalTok{ Tages\_der\_blodproeve\_i\_forbindel }\ControlFlowTok{in}\SpecialCharTok{:}\NormalTok{ (}\StringTok{\textquotesingle{}Ja\textquotesingle{}}\NormalTok{) then do;}
        \ControlFlowTok{if}\NormalTok{ Er\_patienten\_fastende\_}\OtherTok{=}\ErrorTok{:}\StringTok{\textquotesingle{}Ja\textquotesingle{}}\NormalTok{ or NewFastende}\OtherTok{=}\ErrorTok{:}\StringTok{\textquotesingle{}Ja\textquotesingle{}}\NormalTok{ then Faste}\OtherTok{=}\StringTok{\textquotesingle{}Ja\textquotesingle{}}\NormalTok{;}
        \ControlFlowTok{else} \ControlFlowTok{if}\NormalTok{ Er\_patienten\_fastende\_}\OtherTok{=}\ErrorTok{:}\StringTok{\textquotesingle{}Nej\textquotesingle{}}\NormalTok{ or NewFastende}\OtherTok{=}\ErrorTok{:}\StringTok{\textquotesingle{}Nej\textquotesingle{}}\NormalTok{ then Faste}\OtherTok{=}\StringTok{\textquotesingle{}Nej\textquotesingle{}}\NormalTok{;}
\NormalTok{    end;}
    \ControlFlowTok{else} \ControlFlowTok{if}\NormalTok{ Tages\_der\_blodproeve\_i\_forbindel }\ControlFlowTok{in}\SpecialCharTok{:}\NormalTok{ (}\StringTok{\textquotesingle{}Nej\textquotesingle{}}\NormalTok{) then do;}
        \ControlFlowTok{if}\NormalTok{ NewFastende}\OtherTok{=}\ErrorTok{:}\StringTok{\textquotesingle{}Ja\textquotesingle{}}\NormalTok{ then Faste}\OtherTok{=}\StringTok{\textquotesingle{}Ja\textquotesingle{}}\NormalTok{;}
        \ControlFlowTok{else} \ControlFlowTok{if}\NormalTok{ NewFastende}\OtherTok{=}\ErrorTok{:}\StringTok{\textquotesingle{}Nej\textquotesingle{}}\NormalTok{ then Faste}\OtherTok{=}\StringTok{\textquotesingle{}Nej\textquotesingle{}}\NormalTok{;}
\NormalTok{    end;}
    \ControlFlowTok{else} \ControlFlowTok{if}\NormalTok{ Tages\_der\_blodproeve\_i\_forbindel }\ControlFlowTok{in}\NormalTok{ (}\StringTok{\textquotesingle{} \textquotesingle{}}\NormalTok{) then do;}
        \ControlFlowTok{if}\NormalTok{ Er\_patienten\_fastende\_}\OtherTok{=}\ErrorTok{:}\StringTok{\textquotesingle{}Ja\textquotesingle{}}\NormalTok{ or NewFastende}\OtherTok{=}\ErrorTok{:}\StringTok{\textquotesingle{}Ja\textquotesingle{}}\NormalTok{ then Faste}\OtherTok{=}\StringTok{\textquotesingle{}Ja\textquotesingle{}}\NormalTok{;}
        \ControlFlowTok{else} \ControlFlowTok{if}\NormalTok{  Er\_patienten\_fastende\_}\OtherTok{=}\ErrorTok{:}\StringTok{\textquotesingle{}Nej\textquotesingle{}}\NormalTok{ or NewFastende}\OtherTok{=}\ErrorTok{:}\StringTok{\textquotesingle{}Nej\textquotesingle{}}\NormalTok{ then Faste}\OtherTok{=}\StringTok{\textquotesingle{}Nej\textquotesingle{}}\NormalTok{; }
\NormalTok{    end;}
\end{Highlighting}
\end{Shaded}

Note: DD2 plan to make an ``official'' definition of Faste.

Upon enrollment, the individuals are informed to be fasting: no
food/liquid (except water) from 10.00 o'clock the night before. Also,
while fasting, the patient should not take any glucose-regulating drugs.

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\chapter{Data documentation}\label{data-documentation}

\section{biobank.sas7bdat}\label{biobank.sas7bdat}

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.2857}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.2286}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.2429}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.2429}}@{}}
\toprule\noalign{}
\begin{minipage}[b]{\linewidth}\raggedright
Format (var x obs)
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
Id variables
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
Unique key
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
Important dates
\end{minipage} \\
\midrule\noalign{}
\endhead
\bottomrule\noalign{}
\endlastfoot
Wide (48 x 11,381) & CPR, ProjektID & CPR (ProjektID) & VejleDato \\
\end{longtable}

The \texttt{biobank}dataset include analysis results from
``målebiomarkører''. In practice, it includes results from all analyses
not part of the 22 variables from the ``pladebiomarkører'' in the Allan
Vaag grant. A few variables from \texttt{dd2core}
(e.g.~\texttt{reg\_dato} and \texttt{Er\_patienten\_fastende\_}) are
included in the \texttt{biobank} dataset.

Data include analysis results from successful analyses. Not all analyses
are performed for all individuals (missing data), and we don't have more
information about specific analyses (i.e., project, analysis method,
unit/kit, non-successful analyses etc.). In some versions of the
dataset, rows are included for all CPR numbers in the population, even
if no analysis results are available.

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\columnwidth - 12\tabcolsep) * \real{0.1111}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 12\tabcolsep) * \real{0.1389}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 12\tabcolsep) * \real{0.2222}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 12\tabcolsep) * \real{0.1528}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 12\tabcolsep) * \real{0.1528}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 12\tabcolsep) * \real{0.1528}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 12\tabcolsep) * \real{0.0694}}@{}}
\caption{Illustration of the overall data structure. The dataset is in
wide format (48 x 11,381), with CPR or ProjektID as the unique
key.}\tabularnewline
\toprule\noalign{}
\begin{minipage}[b]{\linewidth}\raggedright
Row
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
CPR
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
ProjektID
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
Analysis1
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
Analysis2
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
Analysis3
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
\ldots{}
\end{minipage} \\
\midrule\noalign{}
\endfirsthead
\toprule\noalign{}
\begin{minipage}[b]{\linewidth}\raggedright
Row
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
CPR
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
ProjektID
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
Analysis1
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
Analysis2
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
Analysis3
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
\ldots{}
\end{minipage} \\
\midrule\noalign{}
\endhead
\bottomrule\noalign{}
\endlastfoot
1 & CPR1 & ProjektID1 & num. & num. & & \ldots{} \\
2 & CPR2 & ProjektID2 & num. & & num. & \ldots{} \\
3 & CPR3 & ProjektID3 & & & & \ldots{} \\
4 & CPR4 & ProjektID4 & num. & num. & num. & \ldots{} \\
\ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} &
\ldots{} \\
12,098 & CPR12098 & ProjektID12098 & num. & num. & num. & \ldots{} \\
\end{longtable}

\section{biomark.sas7bdat}\label{biomark.sas7bdat}

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.2740}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.3288}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.1644}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.2329}}@{}}
\toprule\noalign{}
\begin{minipage}[b]{\linewidth}\raggedright
Format (var x obs)
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
Id variables
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
Unique key
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
Important dates
\end{minipage} \\
\midrule\noalign{}
\endhead
\bottomrule\noalign{}
\endlastfoot
Long (9 x 201,243) & CPR, ProjektID, ydernr & CPR*Assay & (Vejledato) \\
\end{longtable}

The \texttt{biomark}dataset include analysis results from the 22
``pladebiomarkører'' in the Allan Vaag grant. No dates are included in
the dataset, however, analyses are performed on the enrollment blood
sample. The dataset include approximately 9,200*22=202,400 rows. In
principle, CPR*Assay should be the unique key, however, some analyses
are performed multiple times per individual (due to dilution in the
analysis).

\begin{longtable}[]{@{}lllllll@{}}
\caption{Illustration of the overall data structure. The dataset is in
long format (9 x 201,243), with CPR*Assay as the unique
key.}\tabularnewline
\toprule\noalign{}
Row & CPR & ProjektID & Assay & Value & Info & \ldots{} \\
\midrule\noalign{}
\endfirsthead
\toprule\noalign{}
Row & CPR & ProjektID & Assay & Value & Info & \ldots{} \\
\midrule\noalign{}
\endhead
\bottomrule\noalign{}
\endlastfoot
1 & CPR1 & ProjektID1 & TNF-a & num. & \ldots{} & \ldots{} \\
2 & CPR1 & ProjektID1 & IL-6 & num. & \ldots{} & \ldots{} \\
3 & CPR1 & ProjektID1 & Ang-Like4 & num. & \ldots{} & \ldots{} \\
\ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} &
\ldots{} \\
22 & CPR1 & ProjektID1 & MPO & num. & \ldots{} & \ldots{} \\
23 & CPR2 & ProjektID2 & TNF-a & num. & \ldots{} & \ldots{} \\
24 & CPR2 & ProjektID2 & IL-6 & num. & \ldots{} & \ldots{} \\
\ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} &
\ldots{} \\
\end{longtable}

\phantomsection\label{refs}
\begin{CSLReferences}{1}{1}
\bibitem[\citeproctext]{ref-Christensen_ClinEpi_2012}
Christensen H, Nielsen JS, Sørensen KM, Melbye M, Brandslund I.
\href{https://doi.org/10.2147/clep.S33042}{New national biobank of the
danish center for strategic research on type 2 diabetes (DD2)}. Clin
Epidemiol. 4:37--42.

\bibitem[\citeproctext]{ref-Gedebjerg_DiabCare_2023}
Gedebjerg A, Bjerre M, Kjaergaard AD, Nielsen JS, Rungby J, Brandslund
I, et al. \href{https://doi.org/10.2337/dc22-1353}{CRP, c-peptide, and
risk of first-time cardiovascular events and mortality in early type 2
diabetes: A danish cohort study}. Diabetes Care. 2023;46(5):1037--45.

\bibitem[\citeproctext]{ref-Gedebjerg_DiabCare_2020}
Gedebjerg A, Bjerre M, Kjaergaard AD, Steffensen R, Nielsen JS, Rungby
J, et al. \href{https://doi.org/10.2337/dc20-0345}{Mannose-binding
lectin and risk of cardiovascular events and mortality in type 2
diabetes: A danish cohort study}. Diabetes Care. 2020;43(9):2190--8.

\bibitem[\citeproctext]{ref-Mor_DiabMetabResRev_2014}
Mor A, Svensson E, Rungby J, Ulrichsen SP, Berencsi K, Nielsen JS, et
al. \href{https://doi.org/10.1002/dmrr.2539}{Modifiable clinical and
lifestyle factors are associated with elevated alanine aminotransferase
levels in newly diagnosed type 2 diabetes patients: Results from the
nationwide DD2 study}. Diabetes Metab Res Rev. 2014;30(8):707--15.

\end{CSLReferences}


\backmatter


\end{document}
