---
title: "DD2 biobank data"
#bibliography: 
#- "`r system('kpsewhich ../../references.bib', intern = TRUE)`"
---

{{< downloadthis ../../files/variable_documentation/DD2_questionnaire.csv dname="DD2_questionnaire" icon="file-earmark-text" label="Variable documentation" type="button" class="btn btn-light">}}

When patients are enrolled in the DD2, blood and a urine samples are collected and stored in the biobank in Vejle. The samples are from DD2 baseline, and future projects can have the samples analyzed if additional analyses are needed - the timing will always be baseline DD2.

Further information about the initial idea about the biobank: @Christensen_ClinEpi_2012.\
The documentation for the biobank can be downloaded here (Danish):{{\< downloadthis ../../files/biobank/DD2_dummy.pdf\
dname="20210809-vejledning-i-dd2-proevehaandtering_kba_hjemmeside" icon="file-earmark-pdf" label="Download" type="button" class="btn btn-link"\>}} <!-- {{\< downloadthis ../../files/biobank/20210809-vejledning-i-dd2-proevehaandtering_kba_hjemmeside.pdf\ --> <!-- dname="20210809-vejledning-i-dd2-proevehaandtering_kba_hjemmeside" icon="file-earmark-pdf" label="Download" type="button" class="btn btn-link"\>}} -->

# Data sources

The data in the biobank are joined from multiple studies and data sets. The individual identifier in the biobank is `ProjektID` and it is unique per CPR number. The project ID should end with the digits -00 (individual). Another variable is `Barcode` which is similar to the project id, but it end with a number (e.g. -12, -19, or -99) and denotes the specific sample for the individual (see documentation above). Currently, the biomarker data includes ProjektID and individual-level biomarker results stored in wide format, but we are currently working on a new setup where data will be stored in long format.

Locally at KEA, the (original) data files are stored in the folder:\
O:\\HE_KEA-DATA-RAW0050\\Main Part\\data\\Input Data Sets\\BioBank

Data files from the 2022/2023 rounds are stored in the folder:\
O:\\HE\\KEA-DATA-RAW0050\\Biomarkører

## Original data

In the first phase of DD2, a lot of biomarkers were analysed for the first 1.053 individuals enrolled in DD2 (some have later opted out). These individuals are marked by the variable `BlodProve1053patients` (based on the CPR numbers in data files `First1053Patients` (same as in `UrinResultatermedRatio2013Nov`)) and are predominantly enrolled in 2011-2012.

These are the data files stored at KEA (files read in locally). File names might have changed since we initially received them.

-   `First1053Patients.txt`: We have probably received the data file in November 2012. It includes n=1.053 CPR numbers and they are most likely everyone in the database at that time. The file includes results from the initial blood samples. There are no dates in the data. The data set includes the variables:

    -   CPR
    -   ProjektID
    -   C-peptid (pmol/L)
    -   GAD (kU/l)
    -   Glucose (n=1.050, mmol/L)
    -   ALAT (n=1.030, U/L)
    -   Hæmolyse (n=1.030)
    -   Icteri (n=1.030)
    -   Lipæmi (n=1.030)
    -   AMYLP (n=1.041, U/L)
    -   CRP (n=1.041, mg/L)

-   `UrinResultatermedRatio2013Nov.txt`: We have probably received the data file in December 2013. It includes n=1.053 CPR numbers (same as in `First1053Patients`) with results from the initial urine samples. The data set includes the variables:

    -   CPR
    -   Barkode (ends with -19)
    -   ALBu_mg_L\_ (n=1.041, mg/L, dated November 19, 20, and 23, 2012)
    -   KREA_mmol_L\_ (n=1.041, mmol/L, dated November 19, 20, and 23, 2012)
    -   UPROT_g\_L\_ (n=1.041, g/L, dated November 19, 20, and 23, 2012)
    -   Albumin_Kreatinin_ratio (n=1.041)
    -   Kommentar (n=473, \<5 with "*PROT-U \> 2.5 g/l*" and the rest with "*ALB-U \<3 mg/L*")

-   `DataTilDD2medFastende.txt`: We have probably received the file in September 2015. It includes n=5.966 CPR numbers with character results on GAD, glucose, and cpeptid, along with a variable about faste. The majority of the individuals in the file are enrolled before 2015, but it only includes around 80-85% of the individuals enrolled before 2016. There are no dates in the data. The data set includes the variables:

    -   Id (ProjectID, ends with -00)
    -   GAD (numeric values for n=131 ID numbers, the value "*\<0,000*" for n=5.797 ID numbers, "*\>525,000*" for n=51 ID numbers, and "*?????*" or "*NA*" for the remaining n=20 ID numbers)
    -   Glucose (numeric values for n=4.457 and "*NA*" for the remaining n=1.539 ID numbers)
    -   Cpeptid (numeric values for n=5.964 and "*NA*" for the remaining n=32 ID numbers)
    -   Fastende ("*Ja*" for n=2.891, "*Nej*" for n=397, and "*Ved ikke*" for the remaining n=2.708 ID numbers) (see below for more infomation about fasting blood samples)

-   `WrongCPeptideMeasurements.txt`: We have probably received the file in November 2015. It includes n=105 ID numbers (end with -00) and the variables cpeptid and NyCpeptid. The values are quite different. We have noted in the syntax that DD2/Jens has told us not to use this file (May 2018).

## Results from Anne Gedebjerg

-   `DD2 resultater.xlsx` (and `DD2 resultater_10_0095`, which is the version where \<10 is replaced by 10 and \<0,095 by 0,095): We have received the file in September 2017 (from Anne Gedebjerg). It includes n=7.519 barcodes (end with -99) in sheets of 100 or 101 rows, and should include CRP and mannose-binding lectin (MBL) on everyone enrolled by December 2016. See @Gedebjerg_DiabCare_2023 for more information. The data set includes the variables:

    -   barkode (ends with -99)
    -   CRP (n=7.510, mg/L)
    -   MBL (n=7.514, µg/L)

-   `Resultater den 250917 Anne Gedebjerg.xlsx`: We have received the file in September 2017 (from Anne Gedebjerg). It includes n=3.116 barcodes (end with -99) and variables regarding MBL expression genotyping (six SNPs in the MBL2 gene). The genotyping was done for the first \~3.000 individuals enrolled in DD2. See @Gedebjerg_DiabCare_2020 for more information. The data set includes the variables:

    -   barkode (ends with -99)
    -   HL
    -   XY
    -   PQ
    -   52
    -   54
    -   57
    -   HAPLOTYPE

<!-- ## New data -->

## Data from April 2022 (upload January 2023)

During 2022-2023 we received additional data on CRP, c-peptid, and glucose. We had a file with CRP in October 2022, but it is fully included in a file from January 2023 which also includes c-peptide and glucose. The October file is therefore not used during uploads.

-   `DD2_cRP_Glucose_Cpep_2022_resultater (1).xlsx`: The data file includes n=3.399 projekt_id. They are all enrolled after the first n=1.053 individuals, but we don't know why they were analysed (not all individuals per year). The data set includes the variables: projekt_id, Cpeptid_Barkode, Cpeptid_Resultat, Cpeptid_Måleenhed, Cpeptid_Antal_decimaler, Cpeptid_Dato, Cpeptid_Notat, CRPHS_Barkode, CRPHS_Resultat, CRPHS_Måleenhed, CRPHS_Antal_decimaler, CRPHS_Dato, CRPHS_Notat, Glukose_Barkode, Glukose_Resultat, Glukose_Måleenhed, Glukose_Antal_decimaler, Glukose_Dato, Glukose_Notat.

    -   projekt_id (n=3.339)
    -   c-peptid (n=2.933, pmol/l, dated 30APR2022 or 01MAY2022)
    -   CRP (n=2.478, mg/l, dated 30APR2022 or 01MAY2022)
    -   Glucose (n=3.055, mmol/L, dated 02APR2022 or 03APR2022)

## HOMA
HOMA values are calculated based on c-peptide and glucose. We use the Oxford calculator which can be downloaded from the website: <https://www.dtu.ox.ac.uk/homacalculator/download.php>


## "Pladebiomarkører"

During 2022-2023, we have received new data from additional biomarker analyses. The majority of the new biomarkers are so-called "pladebiomarkører" (as opposed to the current ones which are called "målebiomarkører") because of the way the analyses are performed. Data in these files are in long format.

-   April 2022, `220211 Vplex_final.xlsx` with 2 sheets: Data received in April 2022 but probably analyzed in February 2022 (no dates in data, but based on date stamps in file names). The data file includes data from n=9.294 DD2 individuals on IL-6 and TNF-a.

    -   Sheet 1, `Vplex sample results_final`. It includes the variables: Sample (id, ends with -12), Sample_Group (=Sample in all rows), Assay (either TNF-a or IL-6), Assay (either TNF-a or IL-6), Calc\_\_Conc\_\_Mean (results), RANGE (value 1 or 2), Plate_Name (each Plate_Name is used 78 times).
        -   TNF-a (n=9.202, pg/ml)
        -   IL-6,(n=9.195, pg/ml)
    -   Sheet 2, `Vplex complete final`. Includes background information about the sample from the sample_groups *Sample* (n=9.024), *Standards* (n=1.888), and *Internal Control* (n=236). The sheet includes the variables Plate_Name, Sample_Group, Sample, Assay, Well, Signal, Mean, CV Calc\_\_Concentration, Calc\_\_Conc\_\_Mean, Calc\_\_Conc\_\_CV, \_\_Recovery, \_\_Recovery_Mean, Detection_Limits\_\_Calc\_\_Low, Detection_Limits\_\_Calc\_\_High, Detection_Range, Detection_Range_yesno, Quantification_range, Quantification_range_yesno, RANGE

    An additional quality log can be downloaded here: {{\< downloadthis ../../files/biobank/DD2_dummy.pdf\
    dname="220215_Overview_Vplex_panel_DD2_sknc" icon="file-earmark-pdf" label="Download" type="button" class="btn btn-link"\>}} <!-- {{\< downloadthis ../../files/biobank/220215_Overview_Vplex_panel_DD2_sknc.pdf\ --> <!-- dname="220215_Overview_Vplex_panel_DD2_sknc" icon="file-earmark-pdf" label="Download" type="button" class="btn btn-link"\>}} --> <!-- I versionen _sknc har jeg fjernet en linje der omhandler et enkelt id-nummer -->

    We also received the data file `DD2 quality log_panel 1 edited_TNF IL6.xlsx` but is has not been used.

-   November 2022, `8plex data final.xlsx` with 9 sheets (overview + 8 biomarkers). We received the data file in November 2022. There are no date stamps indicating when the analyses were performed. The data file include information on n=9.204 individuals (based on sample ID ending with -12). For each assay, the data file includes the variables sample (ID), assay, calc\_\_conc\_\_mean (result), RANGE, and plate_name. In an e-mail, we have been told the unit is pg/ml for all assays.

    -   Ang-Like4 (n=9.200, pg/ml)
    -   FGF-21 (n=9.200, pg/ml)
    -   FGF-23 (Hu FGF-23, n=9.200, pg/ml)
    -   IL1-RA (n=9.200, pg/ml)
    -   Leptin (n=9.196, pg/ml)
    -   RAGE (soluble, n=9.200, pg/ml)
    -   Sclerostin (n=9.200, pg/ml)
    -   U-PAR (n=9.200, pg/ml)

    The overview sheet can be downloaded here: {{\< downloadthis ../../files/biobank/DD2_dummy.pdf\
    dname="8plex_data_final_overview" icon="file-earmark-pdf" label="Download" type="button" class="btn btn-link"\>}}

    We also received data files `eight biomarkers with CPR.xlsx` and `seven biomarkers with CPR.xls` but these have not been used.

-   February 2023, `DD2 osteocalcin final.xlsx` with 3 sheets (overview, results, and *rådata*). We received the file in February 2023, but we have no indication of when the analyses were performed. The file includes n=9.204 individuals (sample, ends with -12). The data file includes the variables sample (ID), assay, calc\_\_conc\_\_mean (result), RANGE, and plate_name.

    -   Osteocalcin-1 (n=9.203)

    The sheet *rådata* includes detailed information about each of the plates. The overview sheet (and additional information from e-mail) can be downloaded here: {{\< downloadthis ../../files/biobank/DD2_dummy.pdf\
    dname="DD2_osteocalcin_final_sknc" icon="file-earmark-pdf" label="Download" type="button" class="btn btn-link"\>}}

-   April 2023, `DD2_7plex_data_final.xlsx` with 9 sheets (overview + 7 biomarkers + additional sheet with sample names). We received the data file in April 2023. There are no date stamps indicating when the analyses were performed. The data file include information on n=9.048 individuals (based on sample ID ending with -12). For each assay, the data file includes the variables sample (ID), assay, calc\_conc\_mean (result), RANGE, and plate_name. In an e-mail, we have been told the unit is pg/ml for all assays. 
    
    -   CD163 (n=9.047, pg/ml)
    -   Galectin-3 (n=9.008, pg/ml)
    -   GDF-15 (n=9.046, pg/ml) 
    -   NT-proBNP (n=9.048, pg/ml)
    -   Resistin (n=9.046, pg/ml) 
    -   Serpin (n=9.047, pg/ml) 
    -   YKL-40 (n=9.045, pg/ml) 



    The overview sheet (and additional information from e-mail) can be downloaded here: {{\< downloadthis ../../files/biobank/DD2_dummy.pdf\
    dname="DD2_7plex_data_final_sknc" icon="file-earmark-pdf" label="Download" type="button" class="btn btn-link"\>}}


-   June 2023, `Osteopontin data final (1).xlsx` with 3 sheets (overview + data + *rådata*). We received the file in June 2023, but we have no indication of when the analyses were performed. This data file has not yet been uploaded to the servers. It includes n=9.207 ID numbers (end with -12, plus a note that it means "EDTA plasma fraction") with data on osteopontin.

    -   Osteopontin (n=9.204, pg/ml)
  
      The overview sheet can be downloaded here: {{\< downloadthis ../../files/biobank/DD2_dummy.pdf\
    dname="Osteopontin data final (1)" icon="file-earmark-pdf" label="Download" type="button" class="btn btn-link"\>}}
  
    
-   "THE LAST ONE" we were told the June 2023 data was "det næstsidste", hence, we expect to receive another data file.


# Fasting 
Is the patient fasting? A simple question, yet, difficult to assess.

Data from the DD2 questionnaire itself includes the variable `Er_patienten_fastende_` (whether the patient is fasting). Currently, around 75% of the individuals have answered that they are fasting. This variable can be used on its own, but should probably be combined with the variable `Tages_der_blodproeve_i_forbindel` (whether the blood sample was taken at the same time as the questionnaire was answered). If the fasting patients are restricted to include only those whose blood sample was taken at the same time as the questionnaire was answered, then around 72% of the individuals are defined to be fasting.  

The data file `DataTilDD2medFastende.txt` received in September 2015 included information on fasting state for n=5.996 individuals ("*Ja*" for n=2.891, "*Nej*" for n=397, and "*Ved ikke*" for n=2.708 individuals). This file will not be updated and we therefore don't get new information on this fasting variable. We don't know how this variable was defined, but it is probably based on information on the blood sample itself. 
By adding the information from this file (variable name `NewFastende`) to the variable `Er_patienten_fastende_`, an additional 447 individuals can be defined as fasting (410 with missing information in `Er_patienten_fastende_` and 37 who replied not to be fasting).

Currently, the fasting state has been defined by the following (SAS) algorithm combining all the files and stating that the patient is fasting if we have any indication that this could be the case (macro: `AdditionalVars_Faste`):

```{r, eval = FALSE}

	if Tages_der_blodproeve_i_forbindel in: ('Ja') then do;
		if Er_patienten_fastende_=:'Ja' or NewFastende=:'Ja' then Faste='Ja';
		else if Er_patienten_fastende_=:'Nej' or NewFastende=:'Nej' then Faste='Nej';
	end;
	else if Tages_der_blodproeve_i_forbindel in: ('Nej') then do;
		if NewFastende=:'Ja' then Faste='Ja';
		else if NewFastende=:'Nej' then Faste='Nej';
	end;
	else if Tages_der_blodproeve_i_forbindel in (' ') then do;
		if Er_patienten_fastende_=:'Ja' or NewFastende=:'Ja' then Faste='Ja';
		else if  Er_patienten_fastende_=:'Nej' or NewFastende=:'Nej' then Faste='Nej'; 
	end;


```




<!-- ------------------------------------------------------------------------ -->

<!-- Få quarto til at markere koden (som når man laver R-koder) - overspringshandling? Koden virker, og jeg har ændret baggrundsfarven (men udkommenteret for at slippe for SAS) -->

<!-- ```{r, class.source="sas_sknc", engine="sas", engine.path="C:/Program Files/SASHome/SASFoundation/9.4/sas.exe"}  -->

<!-- /* brug library(SASmarkdown) fra R */ -->

<!-- proc means data=sashelp.class; -->

<!-- run; -->

<!-- ``` -->

<!-- ```{r} -->

<!-- # tekst -->

<!-- c(1:3) -->

<!-- ``` -->
