% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
  11pt,
  a4paper,
]{book}

\usepackage{amsmath,amssymb}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\ifPDFTeX\else  
    % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\setlength{\emergencystretch}{3em} % prevent overfull lines
\setcounter{secnumdepth}{5}
% Make \paragraph and \subparagraph free-standing
\makeatletter
\ifx\paragraph\undefined\else
  \let\oldparagraph\paragraph
  \renewcommand{\paragraph}{
    \@ifstar
      \xxxParagraphStar
      \xxxParagraphNoStar
  }
  \newcommand{\xxxParagraphStar}[1]{\oldparagraph*{#1}\mbox{}}
  \newcommand{\xxxParagraphNoStar}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
  \let\oldsubparagraph\subparagraph
  \renewcommand{\subparagraph}{
    \@ifstar
      \xxxSubParagraphStar
      \xxxSubParagraphNoStar
  }
  \newcommand{\xxxSubParagraphStar}[1]{\oldsubparagraph*{#1}\mbox{}}
  \newcommand{\xxxSubParagraphNoStar}[1]{\oldsubparagraph{#1}\mbox{}}
\fi
\makeatother


\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
% definitions for citeproc citations
\NewDocumentCommand\citeproctext{}{}
\NewDocumentCommand\citeproc{mm}{%
  \begingroup\def\citeproctext{#2}\cite{#1}\endgroup}
\makeatletter
 % allow citations to break across lines
 \let\@cite@ofmt\@firstofone
 % avoid brackets around text for \cite:
 \def\@biblabel#1{}
 \def\@cite#1#2{{#1\if@tempswa , #2\fi}}
\makeatother
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newlength{\csllabelwidth}
\setlength{\csllabelwidth}{3em}
\newenvironment{CSLReferences}[2] % #1 hanging-indent, #2 entry-spacing
 {\begin{list}{}{%
  \setlength{\itemindent}{0pt}
  \setlength{\leftmargin}{0pt}
  \setlength{\parsep}{0pt}
  % turn on hanging indent if param 1 is 1
  \ifodd #1
   \setlength{\leftmargin}{\cslhangindent}
   \setlength{\itemindent}{-1\cslhangindent}
  \fi
  % set entry spacing
  \setlength{\itemsep}{#2\baselineskip}}}
 {\end{list}}
\usepackage{calc}
\newcommand{\CSLBlock}[1]{\hfill\break\parbox[t]{\linewidth}{\strut\ignorespaces#1\strut}}
\newcommand{\CSLLeftMargin}[1]{\parbox[t]{\csllabelwidth}{\strut#1\strut}}
\newcommand{\CSLRightInline}[1]{\parbox[t]{\linewidth - \csllabelwidth}{\strut#1\strut}}
\newcommand{\CSLIndent}[1]{\hspace{\cslhangindent}#1}

\makeatletter
\@ifpackageloaded{caption}{}{\usepackage{caption}}
\AtBeginDocument{%
\ifdefined\contentsname
  \renewcommand*\contentsname{Table of contents}
\else
  \newcommand\contentsname{Table of contents}
\fi
\ifdefined\listfigurename
  \renewcommand*\listfigurename{List of Figures}
\else
  \newcommand\listfigurename{List of Figures}
\fi
\ifdefined\listtablename
  \renewcommand*\listtablename{List of Tables}
\else
  \newcommand\listtablename{List of Tables}
\fi
\ifdefined\figurename
  \renewcommand*\figurename{Figure}
\else
  \newcommand\figurename{Figure}
\fi
\ifdefined\tablename
  \renewcommand*\tablename{Table}
\else
  \newcommand\tablename{Table}
\fi
}
\@ifpackageloaded{float}{}{\usepackage{float}}
\floatstyle{ruled}
\@ifundefined{c@chapter}{\newfloat{codelisting}{h}{lop}}{\newfloat{codelisting}{h}{lop}[chapter]}
\floatname{codelisting}{Listing}
\newcommand*\listoflistings{\listof{codelisting}{List of Listings}}
\makeatother
\makeatletter
\makeatother
\makeatletter
\@ifpackageloaded{caption}{}{\usepackage{caption}}
\@ifpackageloaded{subcaption}{}{\usepackage{subcaption}}
\makeatother

\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\usepackage{bookmark}

\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same} % disable monospaced font for URLs
\hypersetup{
  pdftitle={Genetic variables},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}


\title{Genetic variables}
\usepackage{etoolbox}
\makeatletter
\providecommand{\subtitle}[1]{% add subtitle to \maketitle
  \apptocmd{\@title}{\par {\large #1 \par}}{}{}
}
\makeatother
\subtitle{Primary data derived from DD2 research studies}
\author{}
\date{}

\begin{document}
\frontmatter
\maketitle

\renewcommand*\contentsname{Table of contents}
{
\setcounter{tocdepth}{2}
\tableofcontents
}

\mainmatter
Go to \href{dd2_genetics.qmd\#data-documentation}{Data documentation}

We received data on genetics/polygenic risk scores (PRS) in multiple
rounds. In the following section we will go through the data files we
received. Of note, we also received PRS data in October 2023 which was
used for one of the birth weight studies (Hansen et al. (2023)), but
these data have not been uploaded to the servers.

On SDS, the first and second rounds of data were originally included in
the \texttt{IL6\_TNF} data, but this data source has now been split into
\texttt{Biomark} including the 22 ``pladebiomarkør'' biomarkers and
\texttt{Genetik} including the genetics data described here. The
variables in \texttt{Genetik} were renamed, i.e.,
\texttt{Assay\ =\ GeneMarker}, \texttt{Calc\_Conc\_Mean\ =\ Value}, and
\texttt{CV\_on\_plate\ =\ data\_round}.

\section{Round 1 (April 2024 - Gen1)}\label{round-1-april-2024---gen1}

Data include 9,997 observations and 104 variables. Some individuals had
missing ID (but potentially non-missing barcode) and some had duplicate
barcodes. The n=215 individuals with missing ID were checked by JSN, and
85 of these were denoted ``ok''. For around 500 individuals, information
on all the PRS values were deleted in the genetic analysis (see README
file for specific reasons). Data were restricted to n=9,856 individuals
with valid, unique, and non-missing ID and barcode, and transformed to
long format. The variables geno\_id, rem, p\_id (text), DD2\_projectid,
DD2\_barcode were removed from data (to align with the format from the
original biobank file, \texttt{IL6\_TNF}). Before upload to SDS
(\texttt{IL6\_TNF\_240411}), 69 individuals were excluded as they were
not among the n=11,381 individuals in the DD2 cohort (SDS, April 2023).
Data were marked with \texttt{data\_round="Gen1"}.

The PRS \texttt{oram\_T1D} has been replaced by \texttt{oram\_T1D\_v2}
in Round 2 as there was an error in the calculation from Round 1. The
variable from Round 1 should therefore not be used.

\section{Round 2 (July 2024 - Gen2)}\label{round-2-july-2024---gen2}

Data include 9,997 observations and 28 variables. Using the same
procedure as in Round 1, we end up with n=9,856 individuals with valid,
unique, and non-missing ID and barcode. Again, data were transformed to
long format, and the variables geno\_id, rem, p\_id (text),
DD2\_projectid, DD2\_barcode were removed. Before upload to SDS
(\texttt{IL6\_TNF\_240711}), 69 individuals were excluded as they were
not among the n=11,381 individuals in the DD2 cohort (SDS, April 2023).
Data were marked with \texttt{data\_round="Gen2"}.

\section{Round 3 (December 2024 -
Gen3)}\label{round-3-december-2024---gen3}

Data include 9,997 observations and 65 variables. Using the same
procedure as in Round 1 and Round 2, we end up with n=9,856 individuals
with valid, unique, and non-missing ID and barcode. Data were marked
with \texttt{data\_round="Gen3"}.

For single SNPs, the original variable name corresponded to
\texttt{chr:position:allele1:allele2\_effect-allele}, however, SAS
replaces \texttt{:} with \texttt{\_} and adds \texttt{\_} in the
beginning of the variable name (as a variable name can't start with a
number), i.e., the variable name is currently
\texttt{\_chr\_position\_allele1\_allele2\_effect-allele}. Additionally,
a document including the r2 values is available on the servers.

\subsection{Round 3, additional data for Tarun
(Tar1)}\label{round-3-additional-data-for-tarun-tar1}

In addition to the Round 3 genetics data, we received a data file
including a broader list with around 1300 SNPs (Tarun Veer Singh
Ahluwalia is lead on the project). The file thus included 9,997
observations and 1309 variables, and we again restricted to the n=9,856
individuals with valid, unique, and non-missing ID and barcode. Data
were marked with \texttt{data\_round="Tar1"}, and variable names
correspond to \texttt{\_chr\_position\_allele1\_allele2\_effect-allele}.
A document including the r2 values is available on the servers.

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\chapter{Data documentation}\label{data-documentation}

\section{genetik.sas7bdat}\label{genetik.sas7bdat}

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.3151}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.2192}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.2329}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.2329}}@{}}
\toprule\noalign{}
\begin{minipage}[b]{\linewidth}\raggedright
Format (var x obs)
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
Id variables
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
Unique key
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
Important dates
\end{minipage} \\
\midrule\noalign{}
\endhead
\bottomrule\noalign{}
\endlastfoot
Long (8 x 14,572,843) & CPR, ProjektID & CPR*GeneMarker & - \\
\end{longtable}

The \texttt{genetik} dataset includes analysis results from the three
data rounds. There are no dates in the dataset, but all analyses are
performed on the enrollment samples. Data can be stratified by round
using the variable \texttt{data\_round} with values \texttt{Gen1},
\texttt{Gen2}, \texttt{Gen3}, and \texttt{Tar1}. A total of 1,489
different gene markers (or PRS or whatever they are) are currently
included in the data. Data include 9,787*1,489=14,572,843 rows.

\begin{longtable}[]{@{}lllllll@{}}
\caption{Illustration of the overall data structure. The dataset is in
long format (8 x 14,572,843), with CPR*GeneMarker as the unique
key.}\tabularnewline
\toprule\noalign{}
Row & CPR & ProjektID & data\_round & GeneMarker & Value & \ldots{} \\
\midrule\noalign{}
\endfirsthead
\toprule\noalign{}
Row & CPR & ProjektID & data\_round & GeneMarker & Value & \ldots{} \\
\midrule\noalign{}
\endhead
\bottomrule\noalign{}
\endlastfoot
1 & CPR1 & ProjektID1 & Gen1 & ACSL1\_rs221 & num. & \ldots{} \\
2 & CPR1 & ProjektID1 & Gen1 & ACSL1\_rs624 & num. & \ldots{} \\
4 & CPR1 & ProjektID1 & Gen1 & ACSL1\_rs756 & num. & \ldots{} \\
\ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} &
\ldots{} \\
1,489 & CPR1 & ProjektID1 & Tar1 & wang\_ST & num. & \ldots{} \\
1,490 & CPR2 & ProjektID2 & Gen1 & ACSL1\_rs221 & num. & \ldots{} \\
1,491 & CPR2 & ProjektID2 & Gen1 & ACSL1\_rs624 & num. & \ldots{} \\
\ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} &
\ldots{} \\
\end{longtable}

Additional documentation and README files (from Mette Andersen, KU) are
available upon request. Among others, they include information about
GeneMarker definition.

\phantomsection\label{refs}
\begin{CSLReferences}{1}{1}
\bibitem[\citeproctext]{ref-Hansen_Diabetologia_2023}
Hansen AL, Thomsen RW, Brøns C, Svane HML, Jensen RT, Andersen MK, et
al. \href{https://doi.org/10.1007/s00125-023-05936-1}{Birthweight is
associated with clinical characteristics in people with recently
diagnosed type 2 diabetes}. Diabetologia. 2023;66(9):1680--92.

\end{CSLReferences}


\backmatter


\end{document}
