<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>DD2 biobank – DD2</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<link href="../../sites/data sources/dd2_genetics.html" rel="next">
<link href="../../sites/data sources/dd2_questionnaire.html" rel="prev">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../sites/data sources/dd2_biobank.html">Additional DD2 data</a></li><li class="breadcrumb-item"><a href="../../sites/data sources/dd2_biobank.html">DD2 biobank</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../../">DD2</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../sites/dd2_enrollment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DD2 enrollment</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../sites/data_structure.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Data structure</span></a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
 <span class="menu-text">The DD2 data collection</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">Primary data</span>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">DD2 enrollment data</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../sites/data sources/dd2_questionnaire.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DD2 questionnaire</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Additional DD2 data</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../sites/data sources/dd2_biobank.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">DD2 biobank</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../sites/data sources/dd2_genetics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Genetic variables</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../sites/data sources/dd2_idnc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Neuropathy questionnaire (IDNC)</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">Secondary data</span>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false">
 <span class="menu-text">External DD2 data</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../sites/data sources/ddda.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DDDA</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../sites/data sources/fodstatus.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Podiatrist data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../sites/data sources/foedsel.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Birth data</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">Routine registry data</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../sites/registry/SDS.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SDS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../sites/registry/DST.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DST</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#data-sources" id="toc-data-sources" class="nav-link active" data-scroll-target="#data-sources">Data sources</a>
  <ul class="collapse">
  <li><a href="#first-biomarkers" id="toc-first-biomarkers" class="nav-link" data-scroll-target="#first-biomarkers">First biomarkers</a></li>
  <li><a href="#additional-biomarkers" id="toc-additional-biomarkers" class="nav-link" data-scroll-target="#additional-biomarkers">Additional biomarkers</a></li>
  <li><a href="#results-from-anne-gedebjerg" id="toc-results-from-anne-gedebjerg" class="nav-link" data-scroll-target="#results-from-anne-gedebjerg">Results from Anne Gedebjerg</a></li>
  <li><a href="#april-2022-data" id="toc-april-2022-data" class="nav-link" data-scroll-target="#april-2022-data">April 2022 data</a></li>
  <li><a href="#c-peptide-and-glucose-full" id="toc-c-peptide-and-glucose-full" class="nav-link" data-scroll-target="#c-peptide-and-glucose-full">C-peptide and glucose (full)</a>
  <ul class="collapse">
  <li><a href="#homa" id="toc-homa" class="nav-link" data-scroll-target="#homa">HOMA</a></li>
  </ul></li>
  <li><a href="#pladebiomarkører" id="toc-pladebiomarkører" class="nav-link" data-scroll-target="#pladebiomarkører">“Pladebiomarkører”</a>
  <ul class="collapse">
  <li><a href="#data-files-pladebiomarkører" id="toc-data-files-pladebiomarkører" class="nav-link" data-scroll-target="#data-files-pladebiomarkører">Data files (pladebiomarkører)</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#fasting" id="toc-fasting" class="nav-link" data-scroll-target="#fasting">Fasting</a></li>
  <li><a href="#data-documentation" id="toc-data-documentation" class="nav-link" data-scroll-target="#data-documentation">Data documentation</a>
  <ul class="collapse">
  <li><a href="#biobank.sas7bdat" id="toc-biobank.sas7bdat" class="nav-link" data-scroll-target="#biobank.sas7bdat">biobank.sas7bdat</a></li>
  <li><a href="#biomark.sas7bdat" id="toc-biomark.sas7bdat" class="nav-link" data-scroll-target="#biomark.sas7bdat">biomark.sas7bdat</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content column-body" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../sites/data sources/dd2_biobank.html">Additional DD2 data</a></li><li class="breadcrumb-item"><a href="../../sites/data sources/dd2_biobank.html">DD2 biobank</a></li></ol></nav>
<div class="quarto-title">
<h1 class="title">DD2 biobank</h1>
<p class="subtitle lead">Primary data derived from DD2 research studies</p>
</div>



<div class="quarto-title-meta column-body">

    
  
    
  </div>
  


</header>


<p>Go to <a href="../../sites/data sources/dd2_biobank.html#data-documentation">Data documentation</a></p>
<!-- ::: callout-important -->
<!-- The website is currently under development. Most of the documentation (enclosed documents) we are linking to are "dummy documents", and we will change this when we have confirmed which documents we are allowed to make publicly available. -->
<!-- ::: -->
<p>When individuals are enrolled in the DD2, blood and urine samples are collected and stored in the biobank in Vejle. The samples themselves are considered “primary DD2 enrollment data”, and they are all collected at DD2 <em>baseline</em>. No automatic or standard analyses are conducted, but DD2 research projects can have the samples analyzed if (additional) analyses are needed - the timing for the analysis results will always be baseline DD2, because it is the time the blood/urine sample was collected. The results from analyses of the blood and urine samples are considered “Additional DD2 data”.</p>
<p>For further information about the initial idea about the biobank, please see <span class="citation" data-cites="Christensen_ClinEpi_2012">Christensen et al.</span><br>
The documentation for the biobank can be downloaded here (Danish):{{&lt; downloadthis ../../files/biobank/vejledning-i-dd2-proevehaandtering_lab_hjemmeside.pdf "vejledning-i-dd2-proevehaandtering_lab_hjemmeside" "file-earmark-pdf" "Download" "button" "btn btn-link" &gt;}}</p>
<!--{{\< downloadthis ../../files/biobank/20210809-vejledning-i-dd2-proevehaandtering_kba_hjemmeside.pdf dname="20210809-vejledning-i-dd2-proevehaandtering_kba_hjemmeside" icon="file-earmark-pdf" label="Download" type="button" class="btn btn-link"\>}}-->
<!-- Husk at fjerne backslash fra start og slut -->
<section id="data-sources" class="level1">
<h1>Data sources</h1>
<p>The data in the biobank are joined from multiple studies and datasets. The individual identifier in the biobank is <code>ProjektID</code> and it is unique per CPR number. The project ID should end with the digits -00 (individual). Another variable is <code>Barcode</code> which is similar to the project id, but it end with a number (e.g.&nbsp;-12, -19, or -99) and denotes the specific sample for the individual (see documentation above).</p>
<p>Locally at DCE, the (original) data files are stored in the folder:<br>
O:\HE_KEA-DATA-RAW0050\Main Part\data\Input Data Sets\BioBank</p>
<p>Data files from the 2022/2023 rounds are stored in the folder:<br>
O:\HE\KEA-DATA-RAW0050\Biomarkører</p>
<p>Below is a list of the biobank data files stored at DCE (files read in locally). File names might have changed since we initially received them.</p>
<section id="first-biomarkers" class="level2">
<h2 class="anchored" data-anchor-id="first-biomarkers">First biomarkers</h2>
<p>In the first phase of DD2, a lot of biomarkers were analysed for the first 1,053 individuals enrolled in DD2 (some individuals have later opted out). These individuals are marked by the variable <code>BlodProve1053patients</code> (based on the CPR numbers in data file <code>First1053Patients</code> (same as in <code>UrinResultatermedRatio2013Nov</code>)) and are predominantly enrolled in 2011-2012.</p>
<ul>
<li><p><code>First1053Patients.txt</code>: The file appears to have been received in November 2012. It includes n=1,053 CPR numbers and these individuals are most likely everyone in the database at that time. The file includes results from the initial blood samples. The dataset lacks date information and also unit specifications for the variables hæmolyse, icteri, and lipæmi. The dataset includes the variables:</p>
<ul>
<li>CPR</li>
<li>ProjektID</li>
<li>C-peptid (n=1,053, pmol/L)</li>
<li>GAD (n=1,053, kU/l)</li>
<li>Glucose (n=1,050, mmol/L)</li>
<li>ALAT (n=1,030, U/L)</li>
<li>Hæmolyse (n=1,030)</li>
<li>Icteri (n=1,030)</li>
<li>Lipæmi (n=1,030)</li>
<li>AMYLP (n=1,041, U/L)</li>
<li>CRP (n=1,041, mg/L)</li>
</ul></li>
</ul>
<p>We have been told (e-mail from JSN 31.05.2018) not to use the c-peptide measurements from this file (old analysis kit).</p>
<p>Information about some of the variables can be found in <span class="citation" data-cites="Mor_DiabMetabResRev_2014">Mor et al. (<a href="#ref-Mor_DiabMetabResRev_2014" role="doc-biblioref">2014</a>)</span>.</p>
<ul>
<li><p><code>UrinResultatermedRatio2013Nov.txt</code>: We likely received the data file in December 2013. It includes n=1,053 CPR numbers (same as in <code>First1053Patients</code>) with results from the initial urine samples. The dataset includes the variables:</p>
<ul>
<li>CPR</li>
<li>Barkode (ends with -19)</li>
<li>ALBu_mg_L_ (n=1,041, mg/L, dated November 19, 20, and 23, 2012)</li>
<li>KREA_mmol_L_ (n=1,041, mmol/L, dated November 19, 20, and 23, 2012)</li>
<li>UPROT_g_L_ (n=1,041, g/L, dated November 19, 20, and 23, 2012)</li>
<li>Albumin_Kreatinin_ratio (n=1,041)</li>
<li>Kommentar (n=473, &lt;5 with “<em>PROT-U &gt; 2.5 g/l</em>” and the rest with “<em>ALB-U &lt;3 mg/L</em>”)</li>
</ul></li>
</ul>
</section>
<section id="additional-biomarkers" class="level2">
<h2 class="anchored" data-anchor-id="additional-biomarkers">Additional biomarkers</h2>
<ul>
<li><p><code>DataTilDD2medFastende.txt</code>: We probably received the file in September 2015 (Anne Gedebjerg). It includes n=5,996 CPR numbers with character results on GAD, glucose, and c-peptide, along with a variable about faste. The majority of the individuals in the file are enrolled before 2015, but it is only around 80-85% of all the individuals enrolled before 2016. There are no dates in the data. The dataset includes the variables:</p>
<ul>
<li>Id (ProjectID, ends with -00)</li>
<li>GAD (numeric values for n=131 ID numbers, the value “<em>&lt;0,000</em>” for n=5,797 ID numbers, “<em>&gt;525,000</em>” for n=51 ID numbers, and “<em>?????</em>” or “<em>NA</em>” for the remaining n=20 ID numbers)</li>
<li>Glucose (numeric values for n=4,457 and “<em>NA</em>” for the remaining n=1,539 ID numbers)</li>
<li>Cpeptid (numeric values for n=5,964 and “<em>NA</em>” for the remaining n=32 ID numbers)</li>
<li>Fastende (“<em>Ja</em>” for n=2,891, “<em>Nej</em>” for n=397, and “<em>Ved ikke</em>” for the remaining n=2,708 ID numbers) (see below for more information about fasting blood samples)</li>
</ul></li>
</ul>
<p>The majority of the initial 1,053 individuals are also included in this data file. We have been told to use the c-peptide measurements in this data file and not the original data file.</p>
<ul>
<li><code>WrongCPeptideMeasurements.txt</code>: The file appears to have been received in November 2015. It includes n=105 ID numbers (end with -00) and the variables cpeptid and NyCpeptid. The values are quite different. We have noted in the syntax that JSN has told us not to use these variables (e-mail 31.05.2018).</li>
</ul>
</section>
<section id="results-from-anne-gedebjerg" class="level2">
<h2 class="anchored" data-anchor-id="results-from-anne-gedebjerg">Results from Anne Gedebjerg</h2>
<ul>
<li><p><code>DD2 resultater.xlsx</code> (and <code>DD2 resultater_10_0095</code>, which is the version where &lt;10 is replaced by 10 and &lt;0,095 by 0,095): We have received the file in September 2017 (from Anne Gedebjerg). It includes n=7,519 barcodes (end with -99) in sheets of 100 or 101 rows, and should include CRP and mannose-binding lectin (MBL) on everyone enrolled by December 2016. See <span class="citation" data-cites="Gedebjerg_DiabCare_2023">Gedebjerg et al. (<a href="#ref-Gedebjerg_DiabCare_2023" role="doc-biblioref">2023</a>)</span> for more information. The dataset includes the variables:</p>
<ul>
<li>barkode (ends with -99)</li>
<li>CRP (n=7,510, mg/L)</li>
<li>MBL (n=7,514, µg/L)</li>
</ul></li>
</ul>
<p>We have been told to keep the CRP measurements from the original file and this file separate. The unit for CRP is mg/L in both data files.</p>
<ul>
<li><p><code>Resultater den 250917 Anne Gedebjerg.xlsx</code>: We have received the file in September 2017 (from Anne Gedebjerg). It includes n=3,116 barcodes (end with -99) and variables regarding MBL expression genotyping (six SNPs in the MBL2 gene). The genotyping was done for the first ~3,000 individuals enrolled in DD2. See <span class="citation" data-cites="Gedebjerg_DiabCare_2020">Gedebjerg et al. (<a href="#ref-Gedebjerg_DiabCare_2020" role="doc-biblioref">2020</a>)</span> for more information. The dataset includes the variables:</p>
<ul>
<li>barkode (ends with -99)</li>
<li>HL</li>
<li>XY</li>
<li>PQ</li>
<li>52</li>
<li>54</li>
<li>57</li>
<li>HAPLOTYPE</li>
</ul></li>
</ul>
<!-- ## New data -->
</section>
<section id="april-2022-data" class="level2">
<h2 class="anchored" data-anchor-id="april-2022-data">April 2022 data</h2>
<p>During 2022-2023 we received additional data on CRP, c-peptide, and glucose. We had a file with CRP in October 2022, but it is fully included in a file from January 2023 which also includes c-peptide and glucose. The October file is therefore not used during uploads, whereas the January 2023 file has been uploaded to the servers.</p>
<ul>
<li><p><code>DD2_cRP_Glucose_Cpep_2022_resultater (1).xlsx</code>: The data file includes n=3,399 projekt_id. They are all enrolled after the first n=1,053 individuals, but we don’t know why they were analysed (not all individuals per year. Maybe in IDA?). The dataset includes the variables: projekt_id, Cpeptid_Barkode, Cpeptid_Resultat, Cpeptid_Måleenhed, Cpeptid_Antal_decimaler, Cpeptid_Dato, Cpeptid_Notat, CRPHS_Barkode, CRPHS_Resultat, CRPHS_Måleenhed, CRPHS_Antal_decimaler, CRPHS_Dato, CRPHS_Notat, Glukose_Barkode, Glukose_Resultat, Glukose_Måleenhed, Glukose_Antal_decimaler, Glukose_Dato, Glukose_Notat.</p>
<ul>
<li>projekt_id (n=3,339)</li>
<li>c-peptid (n=2,933, pmol/l, dated 30APR2022 or 01MAY2022)</li>
<li>CRP (n=2,478, mg/l, dated 30APR2022 or 01MAY2022)</li>
<li>Glucose (n=3,055, mmol/L, dated 02APR2022 or 03APR2022)</li>
</ul></li>
</ul>
</section>
<section id="c-peptide-and-glucose-full" class="level2">
<h2 class="anchored" data-anchor-id="c-peptide-and-glucose-full">C-peptide and glucose (full)</h2>
<p>During the summer 2023 we have received data on c-peptide (July 2023) and glucose (August 2023). These files include <strong>all</strong> measurements from the biobank (cleaned), and results from these files will thus replace all the other measurements from earlier datasets. We now have the files:</p>
<ul>
<li><code>dd2_all_C_peptide_14July2023.xls</code>: Includes n=9,762 project_id with data on c-peptide. The file also includes information about analysis date, sampling date, freezing date, unit, and kit. The following data management has been done by DD2 before we received the file:</li>
</ul>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Quote from e-mail
</div>
</div>
<div class="callout-body-container callout-body">
<!-- e-mail from Sidsel (som i praksis var hende der lavede data management på variablene), 14JUL2023  -->
<pre><code>- Alle 1254 observationer, som havde"gamle" målinger (før 28. feb 2015) er erstattet med opdaterede målinger med nyt kit/assay (efter 28. feb 2015).
- 6 observationer, som kun havde en "gammel" måling og INGEN opdateret genmåling er slettet
- Der er renset op i data, så hvert individ kun fremgår med én måling, som er analyseret vha. nyt kit/assay </code></pre>
</div>
</div>
<p>Please note that c-peptide is now in another unit (it is just the old unit divided by 1000).</p>
<ul>
<li><code>DD2_glukoser_2023_08_23.xlsx</code>: The data file includes n=9,563 projekt_id and information about glucose, units, and dates.</li>
</ul>
<section id="homa" class="level3">
<h3 class="anchored" data-anchor-id="homa">HOMA</h3>
<p><a href="https://www.rdm.ox.ac.uk/about/our-clinical-facilities-and-mrc-units/DTU/software/homa/history" target="_blank">HOMA</a> values are calculated based on c-peptide and glucose. We use the Oxford calculator which can be downloaded from the website: <a href="https://www.dtu.ox.ac.uk/homacalculator/download.php" target="_blank">https://www.dtu.ox.ac.uk/homacalculator/download.php</a>. Since summer 2024, access to the HOMA calculator requires a licence (should be free of charge for “academic researchers”).</p>
<p>HOMA values in the data are calculated based on the c-peptide and glucose measurements received during the summer 2023. HOMA has only been estimated for glucose values in the interval 3.0-25 and c-peptide 0.2-3.5 (because of an updated Oxford HOMA calculator where values out of range cause problems in the excel calculator). Some individuals might therefore have glucose and c-peptide values but no HOMA values in the new data.</p>
</section>
</section>
<section id="pladebiomarkører" class="level2">
<h2 class="anchored" data-anchor-id="pladebiomarkører">“Pladebiomarkører”</h2>
<p>During 2022-2023, we have received new data from additional biomarker analyses (from the Allan Vaag grant). The majority of the new biomarkers are so-called “pladebiomarkører” (as opposed to the current ones which are called “målebiomarkører”) because of the way the analyses are performed. In practice, everyone enrolled as of the day the blood samples were taken from the biobank (in the beginning of 2022) were included in the analysis (approx. the first 9,200 individuals). A small number of individuals have multiple measurements for specific biomarkers, most likely due to sample dilution during the analytical process. Data are in long format and include the following 22 biomarkers (all with unit pg/ml) and the dates refer to when we received the data files:</p>
<ul>
<li>TNF-a (April 2022, n=9,202, pg/ml)</li>
<li>IL-6 (April 2022, n=9,195, pg/ml)</li>
<li>Ang-Like4 (November 2022, n=9,200, pg/ml)</li>
<li>FGF-21 (November 2022, n=9,200, pg/ml)</li>
<li>FGF-23 (November 2022, Hu FGF-23, n=9,200, pg/ml)</li>
<li>IL1-RA (November 2022, n=9,200, pg/ml)</li>
<li>Leptin (November 2022, n=9,196, pg/ml)</li>
<li>RAGE (November 2022, soluble, n=9,200, pg/ml)</li>
<li>Sclerostin (November 2022, n=9,200, pg/ml)</li>
<li>U-PAR (November 2022, n=9,200, pg/ml)</li>
<li>Osteocalcin-1 (February 2023, n=9,203)</li>
<li>CD163 (April 2023, n=9,047, pg/ml)</li>
<li>Galectin-3 (April 2023, n=9,008, pg/ml)</li>
<li>GDF-15 (April 2023, n=9,046, pg/ml)</li>
<li>NT-proBNP (April 2023, n=9,048, pg/ml)</li>
<li>Resistin (April 2023, n=9,046, pg/ml)</li>
<li>Serpin (April 2023, n=9,047, pg/ml)</li>
<li>YKL-40 (April 2023, n=9,045, pg/ml)</li>
<li>Osteopontin (June 2023, n=9,204, pg/ml)</li>
<li>Adiponectin (July 2023, n=9,204, pg/ml)</li>
<li>Follistatin (July 2023, n=9,204, pg/ml)</li>
<li>MPO (July 2023, n=9,204, pg/ml)</li>
</ul>
<p>An overview of the biomarkers (table from the Allan Vaag grant application) can be found here: {{&lt; downloadthis ../../files/biobank/Pladebiomarkorer.pdf "Pladebiomarkorer" "file-earmark-pdf" "Download" "button" "btn btn-link" &gt;}}</p>
<p>An additional document combining overview sheets, method descriptions, and quality logs from some of the analysis rounds can be downloaded here: {{&lt; downloadthis ../../files/biobank/Combined.pdf "Combined_pladebio" "file-earmark-pdf" "Download" "button" "btn btn-link" &gt;}} <!-- I nogle af dokumenterne har jeg fjernet information der omhandler et enkelt id-nummer --></p>
<p>The data files with “pladebiomarkører” and “målebiomarkører” don’t have the same format (e.g., long vs.&nbsp;wide format) and we have thus not combined the data files.</p>
<section id="data-files-pladebiomarkører" class="level3">
<h3 class="anchored" data-anchor-id="data-files-pladebiomarkører">Data files (pladebiomarkører)</h3>
<p>In the following section we will go through the data files we have received.</p>
<ul>
<li><p>April 2022, <code>220211 Vplex_final.xlsx</code> with 2 sheets (data and background information). Data received in April 2022 but probably analyzed in February 2022 (no dates in data, but based on date stamps in file names). The data file includes data from n=9,294 DD2 individuals on IL-6 and TNF-a. The first sheet, <code>Vplex sample results_final</code>, includes the variables: Sample (id, ends with -12), Sample_Group (=Sample in all rows), Assay (either TNF-a or IL-6), Calc__Conc__Mean (results), RANGE (value 1 or 2), Plate_Name (each Plate_Name is used 78 times).</p>
<ul>
<li>TNF-a (n=9,202, pg/ml)</li>
<li>IL-6 (n=9,195, pg/ml)</li>
</ul>
<p>The second sheet, <code>Vplex complete final</code>, includes background information (rådata) about the sample from the sample_groups <em>Sample</em> (n=9,024), <em>Standards</em> (n=1,888), and <em>Internal Control</em> (n=236). The sheet includes the variables Plate_Name, Sample_Group, Sample, Assay, Well, Signal, Mean, CV Calc__Concentration, Calc__Conc__Mean, Calc__Conc__CV, __Recovery, __Recovery_Mean, Detection_Limits__Calc__Low, Detection_Limits__Calc__High, Detection_Range, Detection_Range_yesno, Quantification_range, Quantification_range_yesno, RANGE.</p>
<p>We also received the data file <code>DD2 quality log_panel 1 edited_TNF IL6.xlsx</code> but is has not been used.</p></li>
<li><p>November 2022, <code>8plex data final.xlsx</code>, with 9 sheets (overview + 8 biomarkers). We received the data file in November 2022. There are no date stamps indicating when the analyses were performed. The data file include information on n=9,204 individuals (based on sample ID ending with -12). For each assay, the data file includes the variables sample (ID), assay, calc__conc__mean (result), RANGE, and plate_name. In an e-mail, we have been told the unit is pg/ml for all assays.</p>
<ul>
<li>Ang-Like4 (n=9,200, pg/ml)</li>
<li>FGF-21 (n=9,200, pg/ml)</li>
<li>FGF-23 (Hu FGF-23, n=9,200, pg/ml)</li>
<li>IL1-RA (n=9,200, pg/ml)</li>
<li>Leptin (n=9,196, pg/ml)</li>
<li>RAGE (soluble, n=9,200, pg/ml)</li>
<li>Sclerostin (n=9,200, pg/ml)</li>
<li>U-PAR (n=9,200, pg/ml)</li>
</ul>
<p>We also received data files <code>eight biomarkers with CPR.xlsx</code> and <code>seven biomarkers with CPR.xls</code> but these have not been used.</p></li>
<li><p>February 2023, <code>DD2 osteocalcin final.xlsx</code> with 3 sheets (overview, results, and <em>rådata</em>). We received the file in February 2023, but we have no indication of when the analyses were performed. The file includes n=9,204 individuals (sample, ends with -12). The data file includes the variables sample (ID), assay, calc__conc__mean (result), RANGE, and plate_name.</p>
<ul>
<li>Osteocalcin-1 (n=9,203)</li>
</ul>
<p>The sheet <em>rådata</em> includes detailed information about each of the plates.</p></li>
<li><p>April 2023, <code>DD2_7plex_data_final.xlsx</code> with 9 sheets (overview + 7 biomarkers + additional sheet with sample names). We received the data file in April 2023. There are no date stamps indicating when the analyses were performed. The data file include information on n=9,048 individuals (based on sample ID ending with -12). For each assay, the data file includes the variables sample (ID), assay, calc_conc_mean (result), RANGE, and plate_name. In an e-mail, we have been told the unit is pg/ml for all assays.</p>
<ul>
<li>CD163 (n=9,047, pg/ml)</li>
<li>Galectin-3 (n=9,008, pg/ml)</li>
<li>GDF-15 (n=9,046, pg/ml)</li>
<li>NT-proBNP (n=9,048, pg/ml)</li>
<li>Resistin (n=9,046, pg/ml)</li>
<li>Serpin (n=9,047, pg/ml)</li>
<li>YKL-40 (n=9,045, pg/ml)</li>
</ul></li>
<li><p>June 2023, <code>Osteopontin data final (1).xlsx</code> with 3 sheets (overview + data + <em>rådata</em>). We received the file in June 2023, but we have no indication of when the analyses were performed. It includes n=9,207 ID numbers (end with -12, plus a note that it means “EDTA plasma fraction”) with data on osteopontin.</p>
<ul>
<li>Osteopontin (n=9,204, pg/ml)</li>
</ul></li>
<li><p>July 2023 (1), <code>DD2_adiponectin_blue panel_final (1).xlsx</code> with 3 sheets (overview + data + <em>rådata</em>). We received the file in July 2023, but we have no indication of when the analyses were performed. It includes n=9,204 ID numbers (sample, end with -12) with data on adiponectin.</p>
<ul>
<li>Adiponectin (n=9,204, pg/ml)</li>
</ul>
<p>The sheet <em>rådata</em> includes detailed information about each of the plates.</p></li>
<li><p>July 2023 (2), <code>DD2 red panel_final.xlsx</code> with 4 sheets (overview + data (Follistatin + MPO) + <em>rådata</em>). We received the file in July 2023, but we have no indication of when the analyses were performed. It includes n=9,204 ID numbers (sample, end with -12) with data on follistatin and MPO</p>
<ul>
<li>Follistatin (n=9,204, pg/ml)</li>
<li>MPO (n=9,204, pg/ml)</li>
</ul>
<p>The sheet <em>rådata</em> includes detailed information about each of the plates.</p></li>
</ul>
</section>
</section>
</section>
<section id="fasting" class="level1">
<h1>Fasting</h1>
<p>Is the individual fasting? A simple question, yet, difficult to assess.</p>
<p>Data from the DD2 questionnaire itself include the variable <code>Er_patienten_fastende_</code> (whether the patient is fasting). Currently, around 75% of the individuals have answered that they are fasting. This variable can be used on its own, but should probably be combined with the variable <code>Tages_der_blodproeve_i_forbindel</code> (whether the blood sample was taken at the same time as the questionnaire was answered). If the fasting patients are restricted to include only those whose blood sample was taken at the same time as the questionnaire was answered, then around 72% of the individuals are defined to be fasting.</p>
<p>The data file <code>DataTilDD2medFastende.txt</code> received in September 2015 included information on fasting state for n=5,996 individuals (“<em>Ja</em>” for n=2,891, “<em>Nej</em>” for n=397, and “<em>Ved ikke</em>” for n=2,708 individuals). This file will not be updated and we therefore don’t get new information on this fasting variable. We don’t know how this variable was defined, but it is probably based on information on the blood sample itself. By adding the information from this file (variable name <code>NewFastende</code>) to the variable <code>Er_patienten_fastende_</code>, an additional 447 individuals can be defined as fasting (410 with missing information in <code>Er_patienten_fastende_</code> and 37 who replied not to be fasting in <code>Er_patienten_fastende_</code>).</p>
<p>Currently, the fasting state has been defined by the following (SAS) algorithm combining all the files and stating that the individual is fasting if we have any indication that this could be the case (macro: <code>AdditionalVars_Faste</code>):</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> Tages_der_blodproeve_i_forbindel <span class="cf">in</span><span class="sc">:</span> (<span class="st">'Ja'</span>) then do;</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> Er_patienten_fastende_<span class="ot">=</span><span class="er">:</span><span class="st">'Ja'</span> or NewFastende<span class="ot">=</span><span class="er">:</span><span class="st">'Ja'</span> then Faste<span class="ot">=</span><span class="st">'Ja'</span>;</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span> <span class="cf">if</span> Er_patienten_fastende_<span class="ot">=</span><span class="er">:</span><span class="st">'Nej'</span> or NewFastende<span class="ot">=</span><span class="er">:</span><span class="st">'Nej'</span> then Faste<span class="ot">=</span><span class="st">'Nej'</span>;</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>    end;</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span> <span class="cf">if</span> Tages_der_blodproeve_i_forbindel <span class="cf">in</span><span class="sc">:</span> (<span class="st">'Nej'</span>) then do;</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> NewFastende<span class="ot">=</span><span class="er">:</span><span class="st">'Ja'</span> then Faste<span class="ot">=</span><span class="st">'Ja'</span>;</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span> <span class="cf">if</span> NewFastende<span class="ot">=</span><span class="er">:</span><span class="st">'Nej'</span> then Faste<span class="ot">=</span><span class="st">'Nej'</span>;</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>    end;</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span> <span class="cf">if</span> Tages_der_blodproeve_i_forbindel <span class="cf">in</span> (<span class="st">' '</span>) then do;</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> Er_patienten_fastende_<span class="ot">=</span><span class="er">:</span><span class="st">'Ja'</span> or NewFastende<span class="ot">=</span><span class="er">:</span><span class="st">'Ja'</span> then Faste<span class="ot">=</span><span class="st">'Ja'</span>;</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span> <span class="cf">if</span>  Er_patienten_fastende_<span class="ot">=</span><span class="er">:</span><span class="st">'Nej'</span> or NewFastende<span class="ot">=</span><span class="er">:</span><span class="st">'Nej'</span> then Faste<span class="ot">=</span><span class="st">'Nej'</span>; </span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>    end;</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Note: DD2 plan to make an “official” definition of Faste.</p>
<p>Upon enrollment, the individuals are informed to be fasting: no food/liquid (except water) from 10.00 o’clock the night before. Also, while fasting, the patient should not take any glucose-regulating drugs.</p>
<!-- ::: callout-note -->
<!-- Quote from patientvejledning: "møde fastende til en blodprøvetagning, dvs. ikke have indtaget føde bortset fra vand fra kl. 22.00 aftenen forinden. Desuden må du ikke tage blodsukkerregulerende medicin den tid, du faster." -->
<!-- ::: -->
<!-- ------------------------------------------------------------------------ -->
<!-- Få quarto til at markere koden (som når man laver R-koder) - overspringshandling? Koden virker, og jeg har ændret baggrundsfarven (men udkommenteret for at slippe for SAS) -->
<!-- ```{r, class.source="sas_sknc", engine="sas", engine.path="C:/Program Files/SASHome/SASFoundation/9.4/sas.exe"}  -->
<!-- /* brug library(SASmarkdown) fra R */ -->
<!-- proc means data=sashelp.class; -->
<!-- run; -->
<!-- ``` -->
<!-- ```{r} -->
<!-- # tekst -->
<!-- c(1:3) -->
<!-- ``` -->
<hr>
</section>
<section id="data-documentation" class="level1">
<h1>Data documentation</h1>
<section id="biobank.sas7bdat" class="level2">
<h2 class="anchored" data-anchor-id="biobank.sas7bdat">biobank.sas7bdat</h2>
{{&lt; downloadthis ../../files/variable_documentation/biobank.xlsx "biobank" "file-earmark-text" "Variable documentation" "button" "btn btn-light" &gt;}}
<table class="caption-top table">
<colgroup>
<col style="width: 28%">
<col style="width: 22%">
<col style="width: 24%">
<col style="width: 24%">
</colgroup>
<thead>
<tr class="header">
<th>Format (var x obs)</th>
<th>Id variables</th>
<th>Unique key</th>
<th>Important dates</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Wide (48 x 11,381)</td>
<td>CPR, ProjektID</td>
<td>CPR (ProjektID)</td>
<td>VejleDato</td>
</tr>
</tbody>
</table>
<p>The <code>biobank</code>dataset include analysis results from “målebiomarkører”. In practice, it includes results from all analyses not part of the 22 variables from the “pladebiomarkører” in the Allan Vaag grant. A few variables from <code>dd2core</code> (e.g.&nbsp;<code>reg_dato</code> and <code>Er_patienten_fastende_</code>) are included in the <code>biobank</code> dataset.</p>
<p>Data include analysis results from successful analyses. Not all analyses are performed for all individuals (missing data), and we don’t have more information about specific analyses (i.e., project, analysis method, unit/kit, non-successful analyses etc.). In some versions of the dataset, rows are included for all CPR numbers in the population, even if no analysis results are available.</p>
<table class="caption-top table">
<caption>Illustration of the overall data structure. The dataset is in wide format (48 x 11,381), with CPR or ProjektID as the unique key.</caption>
<colgroup>
<col style="width: 11%">
<col style="width: 13%">
<col style="width: 22%">
<col style="width: 15%">
<col style="width: 15%">
<col style="width: 15%">
<col style="width: 6%">
</colgroup>
<thead>
<tr class="header">
<th>Row</th>
<th>CPR</th>
<th>ProjektID</th>
<th>Analysis1</th>
<th>Analysis2</th>
<th>Analysis3</th>
<th>…</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>CPR1</td>
<td>ProjektID1</td>
<td>num.</td>
<td>num.</td>
<td></td>
<td>…</td>
</tr>
<tr class="even">
<td>2</td>
<td>CPR2</td>
<td>ProjektID2</td>
<td>num.</td>
<td></td>
<td>num.</td>
<td>…</td>
</tr>
<tr class="odd">
<td>3</td>
<td>CPR3</td>
<td>ProjektID3</td>
<td></td>
<td></td>
<td></td>
<td>…</td>
</tr>
<tr class="even">
<td>4</td>
<td>CPR4</td>
<td>ProjektID4</td>
<td>num.</td>
<td>num.</td>
<td>num.</td>
<td>…</td>
</tr>
<tr class="odd">
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
</tr>
<tr class="even">
<td>12,098</td>
<td>CPR12098</td>
<td>ProjektID12098</td>
<td>num.</td>
<td>num.</td>
<td>num.</td>
<td>…</td>
</tr>
</tbody>
</table>
</section>
<section id="biomark.sas7bdat" class="level2">
<h2 class="anchored" data-anchor-id="biomark.sas7bdat">biomark.sas7bdat</h2>
{{&lt; downloadthis ../../files/variable_documentation/biomark.xlsx "biomark" "file-earmark-text" "Variable documentation" "button" "btn btn-light" &gt;}}
<table class="caption-top table">
<colgroup>
<col style="width: 27%">
<col style="width: 32%">
<col style="width: 16%">
<col style="width: 23%">
</colgroup>
<thead>
<tr class="header">
<th>Format (var x obs)</th>
<th>Id variables</th>
<th>Unique key</th>
<th>Important dates</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Long (9 x 201,243)</td>
<td>CPR, ProjektID, ydernr</td>
<td>CPR*Assay</td>
<td>(Vejledato)</td>
</tr>
</tbody>
</table>
<p>The <code>biomark</code>dataset include analysis results from the 22 “pladebiomarkører” in the Allan Vaag grant. No dates are included in the dataset, however, analyses are performed on the enrollment blood sample. The dataset include approximately 9,200*22=202,400 rows. In principle, CPR*Assay should be the unique key, however, some analyses are performed multiple times per individual (due to dilution in the analysis).</p>
<table class="caption-top table">
<caption>Illustration of the overall data structure. The dataset is in long format (9 x 201,243), with CPR*Assay as the unique key.</caption>
<thead>
<tr class="header">
<th>Row</th>
<th>CPR</th>
<th>ProjektID</th>
<th>Assay</th>
<th>Value</th>
<th>Info</th>
<th>…</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>CPR1</td>
<td>ProjektID1</td>
<td>TNF-a</td>
<td>num.</td>
<td>…</td>
<td>…</td>
</tr>
<tr class="even">
<td>2</td>
<td>CPR1</td>
<td>ProjektID1</td>
<td>IL-6</td>
<td>num.</td>
<td>…</td>
<td>…</td>
</tr>
<tr class="odd">
<td>3</td>
<td>CPR1</td>
<td>ProjektID1</td>
<td>Ang-Like4</td>
<td>num.</td>
<td>…</td>
<td>…</td>
</tr>
<tr class="even">
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
</tr>
<tr class="odd">
<td>22</td>
<td>CPR1</td>
<td>ProjektID1</td>
<td>MPO</td>
<td>num.</td>
<td>…</td>
<td>…</td>
</tr>
<tr class="even">
<td>23</td>
<td>CPR2</td>
<td>ProjektID2</td>
<td>TNF-a</td>
<td>num.</td>
<td>…</td>
<td>…</td>
</tr>
<tr class="odd">
<td>24</td>
<td>CPR2</td>
<td>ProjektID2</td>
<td>IL-6</td>
<td>num.</td>
<td>…</td>
<td>…</td>
</tr>
<tr class="even">
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
</tr>
</tbody>
</table>



</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-Christensen_ClinEpi_2012" class="csl-entry" role="listitem">
Christensen H, Nielsen JS, Sørensen KM, Melbye M, Brandslund I. <a href="https://doi.org/10.2147/clep.S33042">New national biobank of the danish center for strategic research on type 2 diabetes (DD2)</a>. Clin Epidemiol. 4:37–42.
</div>
<div id="ref-Gedebjerg_DiabCare_2023" class="csl-entry" role="listitem">
Gedebjerg A, Bjerre M, Kjaergaard AD, Nielsen JS, Rungby J, Brandslund I, et al. <a href="https://doi.org/10.2337/dc22-1353">CRP, c-peptide, and risk of first-time cardiovascular events and mortality in early type 2 diabetes: A danish cohort study</a>. Diabetes Care. 2023;46(5):1037–45.
</div>
<div id="ref-Gedebjerg_DiabCare_2020" class="csl-entry" role="listitem">
Gedebjerg A, Bjerre M, Kjaergaard AD, Steffensen R, Nielsen JS, Rungby J, et al. <a href="https://doi.org/10.2337/dc20-0345">Mannose-binding lectin and risk of cardiovascular events and mortality in type 2 diabetes: A danish cohort study</a>. Diabetes Care. 2020;43(9):2190–8.
</div>
<div id="ref-Mor_DiabMetabResRev_2014" class="csl-entry" role="listitem">
Mor A, Svensson E, Rungby J, Ulrichsen SP, Berencsi K, Nielsen JS, et al. <a href="https://doi.org/10.1002/dmrr.2539">Modifiable clinical and lifestyle factors are associated with elevated alanine aminotransferase levels in newly diagnosed type 2 diabetes patients: Results from the nationwide DD2 study</a>. Diabetes Metab Res Rev. 2014;30(8):707–15.
</div>
</div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation column-body">
  <div class="nav-page nav-page-previous">
      <a href="../../sites/data sources/dd2_questionnaire.html" class="pagination-link" aria-label="DD2 questionnaire">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">DD2 questionnaire</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../../sites/data sources/dd2_genetics.html" class="pagination-link" aria-label="Genetic variables">
        <span class="nav-page-text">Genetic variables</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
      <ul class="footer-items list-unstyled">
    <li class="nav-item">
 SKNC
  </li>  
</ul>
    </div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>




</body></html>